---
title: "Peptide_cleavage_data_analysis"
output: html_document
---


This is analysis and data to support 
"Alternative proteoforms and proteoform-dependent assemblies in humans and plants"



# Load libraries and functions and set style
```{r libraries}
library(tidyverse)
library(cowplot)
library(broom)
library(patchwork)
library(ggridges)
#library(seqinr) # Not working on marccomp?
library(future)
library(furrr)
library(plotly)
library(shiny)

source("scripts/fasta_utils.R")
source("scripts/theme_utils.R")
source("scripts/saved_ggplots.R")
source("scripts/simple_AdaptGauss.R")
theme_set(theme_cowplot_consistent_text(font_size = 8))
palette <- rep(c("#0072B2","#E69F00","#009E24","#FF0000", "#979797","#5530AA"), 8)
```


Files made in this script (for making figures without have to rerun things)
```{r}



short_tidy_unique <- read_csv("data_files/short_tidy_unique.csv")
peaks <- read_csv("/project/cmcwhite/data/peptide_elutions/peaks_rms.csv")
bps_identified <-  read_csv("data/bps_identified.csv")
fragment_data_terminal <- read_csv("/project/cmcwhite/data/fragment_data_terminal.csv")



human_proteome <- read_csv("data_files/human_proteome.csv")
arath_proteome <- read_csv("data_files/arath_proteome.csv")
chlre_proteome <- read_csv("data_files/chlre_proteome.csv")
meta <- read_csv("/project/cmcwhite/data/peptide_elutions/annotation_files/meta.csv") 
domains_pfam <- read_csv("/project/cmcwhite/data/peptide_elutions/annotation_files/domains_pfam.csv")
orthologies <- read_csv("/project/cmcwhite/data/peptide_elutions/orthologies.csv")
domaindis_setup <- read_tsv("data_files/domaindis_setup.txt")


labels <- read_csv("labels.csv")




```


# Load and format proteomics data

### Load peptide elution data
```{r}

# Get those arabidopsis IEX's
data_path <- "data"
short_tidy_filenames <- dir(data_path, pattern = ".*.pepcount.annot.short.tidy$") # Get filenames of data


# Data format description

#Peptide,ExperimentID,experiment_order,experiment_name,spec,ProteinID,Start,End,status,ID,seqlen,FractionID,pepcount,FractionOrder,totfracts
#MDMEMK,soybn_Soy_SECwithStandards,20,SOYBN_SEC2,soybn,tr|I1JP35|I1JP35_SOYBN,1,6,protein_unique,KOG4249,419,SoySEC_standards_03a_03142018,2,2,70

# Minimum necessary columns are Peptide, ExperimentID, ProteinID, Start, End, FractionOrder, pepcount
# Start,End give location of peptide, indexed from 1
# FractionOrder (numeric) is which fraction where the peptide was seen

short_tidy_filenames


short_tidy <- tibble(filename = short_tidy_filenames) %>%
               mutate(file_contents = map(filename, ~ read_csv(file.path(data_path, .)))) %>%
               unnest(cols = c(file_contents)) %>%
  as_tibble() %>%
  rename(totfracs = totfracts)

# Only take proteins with at least 10 unique peptides observed across experiments
numpeps_filter <- short_tidy %>%
  select(ProteinID, Peptide) %>%
  unique() %>%
  dplyr::count(ProteinID) %>% #seqinr can overwrite count
  filter(n >= 10) %>%
  select(ProteinID)

short_tidy %>% write_csv("data/short_tidy.csv")
short_tidy <- read_csv("data/short_tidy.csv")

numpeps_filter %>% write_csv("data/numpeps_filter_10.csv")
numpeps_filter <- read_csv("data/numpeps_filter_10.csv")


short_tidy_filt <- short_tidy %>%
   filter(ProteinID %in% numpeps_filter$ProteinID)

```


### Get only protein unique peptides
```{r}

short_tidy_unique <- short_tidy_filt %>%
  filter(status == "protein_unique") %>% select(-filename, -status)
short_tidy_unique %>% write_csv("data_files/short_tidy_unique.csv")

short_tidy_unique <- read_csv("data_files/short_tidy_unique.csv")
```






# Load accessory files


### Load proteome sequences and annotations
```{r}


human_proteome <- read_fasta('/project/cmcwhite/data/peptide_elutions/protein_identification/proteomes/human_uniprot-proteome_human_reviewed.fasta')

arath_proteome <- read_fasta('/project/cmcwhite/data/peptide_elutions/protein_identification/proteomes/arath_uniprot-proteome_UP000006548.fasta')

chlre_proteome <- read_fasta('/project/cmcwhite/data/peptide_elutions/protein_identification/proteomes/chlre_uniprot-proteome_UP000006906.fasta')



human_proteome %>% write_csv("data_files/human_proteome.csv")
arath_proteome %>% write_csv("data_files/arath_proteome.csv")
chlre_proteome %>% write_csv("data_files/chlre_proteome.csv")

human_proteome <- read_csv("data_files/human_proteome.csv")
arath_proteome <- read_csv("data_files/arath_proteome.csv")
chlre_proteome <- read_csv("data_files/chlre_proteome.csv")

get_meta <- function(proteome){
  meta <- proteome %>%
  select(ID, Sequence) %>%
   separate(ID, into = c(NA, 'Entry', 'Entry_name'), sep = "[|]") %>%
  mutate(seqlen = str_length(Sequence)) %>%
    
  select(-Sequence)
  return(meta)
  
}
human_meta <- get_meta(human_proteome)
arath_meta <- get_meta(arath_proteome)
chlre_meta <- get_meta(chlre_proteome)


meta <- bind_rows(
  human_meta,
  arath_meta,
  chlre_meta
) 

meta %>% write_csv("/project/cmcwhite/data/peptide_elutions/data_files/meta.csv") 

```





### Orthologies for consolidation related proteins across species
```{r}
human_orthology <- read_tsv("protein_identification/eggnog_mapping/human_hmmer_euNOG.mapping")
arath_orthology <- read_tsv("protein_identification/eggnog_mapping/arath_hmmer_euNOG.mapping")
chlre_orthology <- read_tsv("protein_identification/eggnog_mapping/chlre_hmmer_euNOG.mapping")


orthologies <-
  bind_rows(human_orthology,
            arath_orthology,
            chlre_orthology)

orthologies %>% write_csv("/project/cmcwhite/data/peptide_elutions/orthologies.csv")
```


### For comparing fragments to known isoforms
```{r}
human_w_isoform <- read_fasta("annotation_files/uniprot_human_reviewed_isoforms.fasta")

human_w_isoform %>%
   separate(ID, into = c("repo", "Entry_tmp", "Entry_name"), remove = FALSE, sep = "[|]") %>%
  separate(Entry_tmp, into = c("Entry", "isoform_num"), sep = "-") %>%
  group_by(Entry) %>%
     summarize(n = n()) %>%
  ungroup %>%
  mutate(isoform_status = case_when(n == 1 ~ "no_isoform",
                                    n >1 ~ "isoforms")) %>%
  group_by(isoform_status) %>%  
      summarise(n = n())

# Proteins with isoform
#10535/(9817 + 10535)
```

# Load domain/disorder coordinate annotations
### Downloaded from https://mobidb.org/ browse
### Domain coordinate annotations

```{r}

domains_human <- read_delim("/project/cmcwhite/data/exons/data/human_reviewed_accs_interpro.txt", delim ="\t", col_names = c("Entry", "Interpro_ID", "Annotation",  "Source_ID", "ProtBegin", "ProtEnd"))
domains_pfam_human <- domains_human %>% filter(grepl("^PF", Source_ID))

domains_arath <- read_delim("/project/cmcwhite/data/exons/data/arath_accs_interpro.txt", delim ="\t", col_names = c("Entry", "Interpro_ID", "Annotation",  "Source_ID", "ProtBegin", "ProtEnd"))
domains_pfam_arath <- domains_arath %>% filter(grepl("^PF", Source_ID))

domains_chlre <- read_delim("/project/cmcwhite/data/exons/data/chlre_accs_interpro.txt", delim ="\t", col_names = c("Entry", "Interpro_ID", "Annotation",  "Source_ID", "ProtBegin", "ProtEnd"))
domains_pfam_chlre <- domains_chlre %>% filter(grepl("^PF", Source_ID))

domains_pfam <- bind_rows(
  domains_pfam_human,
  domains_pfam_arath,
  domains_pfam_chlre
)

domains_pfam %>% write_csv("/project/cmcwhite/data/peptide_elutions/annotation_files/domains_pfam.csv")

```

```{r}

mobidb_human <- read_delim("/project/cmcwhite/data/exons/data/mobidb_human.txt", delim ="\t") 
mobidb_arath <- read_delim("/project/cmcwhite/data/exons/data/mobidb_arath.txt", delim ="\t") 
mobidb_chlre <- read_delim("/project/cmcwhite/data/exons/data/mobidb_chlre.txt", delim ="\t") 

domains_dis <- bind_rows(
  mobidb_chlre,
  mobidb_human,
  mobidb_arath
) %>%
  filter(feature %in%  c("prediction-disorder-mobidb_lite", "prediction-transmembrane-uniprot")) %>%
  separate_rows(`start..end`, sep = ",") %>%
  separate(`start..end`, into = c("ProtBegin", "ProtEnd"), sep = "[.][.]") %>%
  mutate(ProtBegin = as.integer(ProtBegin)) %>%
  mutate(ProtEnd = as.integer(ProtEnd)) %>%
  rename(Entry = acc) %>%
  select(Entry, ProtBegin, ProtEnd, feature) %>% 
  mutate(set = case_when(feature == "prediction-disorder-mobidb_lite"~ "Disordered",
                              feature == "prediction-transmembrane-uniprot" ~ "Transmembrane")) %>%
  select(-feature)
```

Make annotation plots
```{r}
domains_dis

domains_dis_formatted <- domains_dis%>%
  select(Entry, ProtBegin,ProtEnd, set) %>%
  mutate(Annotation ="Disordered") %>%
  
  mutate(ymin = case_when(set == "Disordered" ~ 0.8,
                          set == "Transmembrane" ~ 1)) %>%
   mutate(ymax = case_when(set == "Disordered" ~ 1,
                          set == "Transmembrane" ~ 1.4))

domains_pfam_formatted <- domains_pfam %>%
  select(Entry, ProtBegin,ProtEnd, Annotation) %>%
  mutate(ymin = 0.1, ymax = 0.8) %>%
  mutate(set = 'Domains')

domaindis_setup <- 
  bind_rows(
    domains_pfam_formatted,
    domains_dis_formatted
    
  ) %>%
  full_join(meta) %>%
  mutate(set = fct_relevel(set, c("Domains", "Disordered", "Transmembrane"))) %>%
  filter(!is.na(Entry_name)) # There are some other proteins brought in with pfam

domaindis_setup %>%
  write_tsv("data_files/domaindis_setup.txt")

domaindis_setup <-
  read_tsv("data_files/domaindis_setup.txt")

domaindis_setup
domaindis_setup %>%
 # filter(Entry_name == "RK2_CHLRE") %>%
 filter(Entry_name == "MER34_HUMAN") %>%
  ggplot() +
  geom_rect(fill = "grey90", aes(xmin = 0, xmax = seqlen, ymin = 0.1, ymax = 0.8)) +
  geom_rect(aes(xmin = ProtBegin, xmax = ProtEnd, ymin = ymin, ymax = ymax, fill = set)) +
  theme(axis.line.x = element_blank(),
        axis.text.x = element_blank(),
        axis.ticks.x= element_blank(),
        axis.line.y = element_blank(),
        axis.title.y = element_blank()) +
  coord_flip() +
  scale_fill_manual(values = c("orange", "cadetblue4", "purple")) +
  scale_y_continuous(limits = c(0,15), expand = c(0,0)) +
  scale_x_reverse(expand = c(0,0)) +
  #scale_y_continuous()
  geom_text_repel(y = 1, ylim = c(1,15), aes(x = ProtBegin, label = Annotation)) +
  theme(legend.position = "none") 
  
  

```



```{r}

domaindis_setup <- read_tsv("data_files/domaindis_setup.txt")

  domain_dis_plot <- domaindis_setup %>%
  filter(Entry_name == "WDR44_HUMAN") %>%
    mutate(label = str_replace(Entry_name, "_", "\n")) %>%
    mutate(label = str_replace(label, "A0A2K3", "c")) %>%
  
  ggplot() +
  geom_rect(fill = "grey90", aes(xmin = 0, xmax = seqlen, ymin = 0.1, ymax = 0.8)) +
  geom_rect(aes(xmin = ProtBegin, xmax = ProtEnd, ymin = ymin, ymax = ymax, fill = set)) +
  theme(axis.line.y = element_blank(),
        axis.text.y = element_blank(),
        axis.ticks.y= element_blank(),
        axis.line.x = element_blank(),
        axis.title.x = element_blank()) +
  scale_fill_manual(values = c("orange", "cadetblue4")) +
  scale_y_continuous(limits = c(0,2), expand = c(0,0)) +
  #scale_x_reverse(expand = c(0,0)) +
  geom_text_repel(alpha = 0.7, y = 1, ylim = c(1,2), size = 2.5, aes(x = ProtBegin, label = Annotation)) +
  theme(legend.position = "none") +
    facet_wrap(~label, ncol = 1, strip.position = "left")

domain_dis_plot



```




# Descriptive files
### Make descriptive files
```{r}
total_exps <-  short_tidy_filt %>%
  select(experiment_name ) %>%
  unique 

fraction_names  <- short_tidy_filt %>%
  select(experiment_name, FractionID, FractionOrder) %>%
  unique 

fraction_names %>%
  write_csv("annotation_files/fraction_names.csv")


total_fractions <- fraction_names %>%
  group_by(experiment_name) %>%
     summarize(totfracs = max(FractionID))
     
total_fractions %>%
  write_csv("annotation_files/total_fractions.csv")


short_tidy_filt %>%
    filter(spec != "soybn") %>%
  summarize(npeps = sum(pepcount))

short_tidy_filt %>%
    filter(spec != "soybn") %>%
  select(ProteinID) %>%
  unique() %>%
  nrow()
 short_tidy_filt
 
 short_tidy_filt

```


# Make small postage stamp plots for visual screening
Step one for picking out positives
```{r}

plan(multiprocess)


make_filename <- function(data, device){
  orthoid <- data$ID[[1]]
  protid <- str_replace_all(data$ProteinID[[1]], "[|]", "_") 
  
  filename <-  paste0(orthoid, "-", protid, ".", device)
  
  return(filename)
}

plotter_wrapper <- function(data, device){
  plt1 <- pep_heatmap_fxn(data)
  plt2 <- pepcov_fxn(data, pubtheme = FALSE)

  plt <- plot_grid(plt1, plt2, align = "vh", axis = "x")
  filename <- make_filename(data, device)
  print(filename)
  ggsave(filename, plt, device = device,  width = 7, height = 4, units = "in")
  return(data.frame(status = "SUCCESS"))
}


safe_plotter <- possibly(plotter_wrapper, otherwise = data.frame(status = "FAILED"))

short_tidy_unique %>% 
  arrange(Start, End) %>%
    filter(ProteinID == "sp|A6NCN2|KR87P_HUMAN") %>%
  split(.$ProteinID) %>%
  future_map_dfr(~plotter_wrapper(.x, "png"), .id = "ProteinID.exp", .progress = TRUE) %>%
  #future_map_dfr(~safe_plotter(.x, "png"), .id = "ProteinID.exp", .progress = TRUE) %>%
  data.frame() -> command_out
```

# Make positive/negative sets



```{r}

# Positive and negative labels were for figuring score. 
# Positives were proteins with interesting elution patterns, negatives were random

set.seed(42)

# Positives were hand selected
positives <- read_csv("data_files/positive_filenames.txt", col_names = "filename") %>% 
  separate(filename, into =c("spec", "experiment_type", "ProteinID"), sep = "_", extra= "merge") %>%
  mutate(ProteinID = str_replace(ProteinID, '.tiff', '')) %>%
  
  mutate(experiment_name = paste0(spec, "_", experiment_type))%>%
 # separate(ProteinID, into = c(NA,"Entry", NA,  NA, NA)) %>%
  mutate(ProteinID = str_replace(ProteinID, "_", "|")) %>%
    mutate(ProteinID = str_replace(ProteinID, "_", "|")) %>%

  select(ProteinID, experiment_name) %>%
    mutate(label = "pos")%>%
  filter(!grepl("HEKU", experiment_name)) %>%
  filter(!grepl("MAIZE_SEC", experiment_name)) %>%
  filter(!grepl("HEKR_IEX", experiment_name)) 


set.seed(43)
positives_rand <- positives %>% 
  sample_n(nrow(positives), replace = F)

positives_train <- positives_rand %>%
   head(as.integer(0.5*nrow(positives_rand))) %>%
  mutate(set = "train")

positives_validate <- positives_rand %>%
  anti_join(positives_train) %>%
  mutate(set = "val")

#positives_train %>%
#  write_csv("data_files/positive_train.csv")
#positives_validate %>%
#  write_csv("data_files/positive_validate.csv")

# Make sure proteins I've been calibrating with are in training set. 
# Expand to more proteins
positives_train %>% filter(grepl("TSS", ProteinID))
positives_train %>% filter(grepl("VILI4", ProteinID))



negatives <- short_tidy  %>%
    
    filter(!ProteinID %in% positives$ProteinID) %>%
  sample_n(1000, replace = F) %>%
  select(ProteinID, experiment_name) %>% unique %>%
  mutate(label = "neg") %>%
    filter(!(ProteinID %in% positives$ProteinID)) 


negatives_train <- negatives %>%
   head(500) %>%
   mutate(set ="train")

negatives_validate <- negatives  %>%
  tail(500) %>%
  mutate(set = "val")


#Alternative negatives 

pos_stats <- positives %>%
  left_join(peptide_stats, by = c("ProteinID", "experiment_name")) %>%
  select(ProteinID, experiment_name, num_counts) %>% mutate(label = "pos") 
  

neg_stats <- peptide_stats %>%
  select(ProteinID, experiment_name, num_counts) %>%
    filter(!ProteinID %in% positives$ProteinID) %>%
    mutate(label = "neg")  
both_stats <- neg_stats %>%
  bind_rows(pos_stats) %>%
  filter(experiment_name != "HEKB_SEC1") %>%
  mutate(quantile = ntile(num_counts, 100))

pos_stats_w_ntile <- both_stats %>% filter(label == "pos")
neg_stats_w_ntile <- both_stats %>% filter(label == "neg")

alt_negatives <-pos_stats_w_ntile %>% left_join(neg_stats_w_ntile, by = c("quantile", "experiment_name"), suffix = c("_pos", "_neg"), multiple = "all") %>%
  group_by(ProteinID_pos, experiment_name) %>%
  slice(1) %>%
  ungroup %>%
  select(ProteinID = ProteinID_neg, experiment_name) %>% mutate(label = "neg")




alt_labels <- bind_rows(alt_negatives, positives)


negatives_train <- negatives %>%
   head(500) %>%
   mutate(set ="train")

negatives_validate <- negatives  %>%
  tail(500) %>%
  mutate(set = "val")



#negatives_train %>%
#  write_csv("data_files/negative_train.csv")
#negatives_validate %>%
#  write_csv("data_files/negative_validate.csv")


labels <- bind_rows(
  positives,#_train,
 # positives_validate,
  negatives
)


labels %>% write_csv("labels.csv")
labels <- read_csv("labels.csv")




```

### Get smaller dataset of just labeled proteins
```{r}

short_tidy_unique_labeled <- short_tidy_unique %>% 
        filter(ProteinID %in% labels$ProteinID) 

```


VILI4_ARATH, TSS_ARATH were used to calibrate gaussian git and downstream to log2fold change

RFA1_HUMAN = has an N-terminal

# Detection of proteoforms
### Start proteoform detection process

Modified AdaptGauss to fit up to 5 peaks, and run for more iterations (finds better fit)

Returns lowest RMS peak fitting

Don't penalize more peaks, since it's legitimate to have one peak or multiple peaks 

Use the output of 

Then move from N->C to look for max difference between coverage in Nterm vs Cterm for each peak

variation 1 -> N

# Part 1: AdaptGauss
##### Testing AdaptGauss prior to run on full set
For one at a time
This is for testing out modifications to the AdaptGauss
```{r}

#testing AdaptGauss process on single protein

# Filter dataset to just one protein in one experiment
tester <- short_tidy_unique_peaks %>% 
   #filter(grepl("VILI4_ARATH", ProteinID)) %>% 
   #filter(experiment_name == "ARATH_IEX1")
   #filter(ProteinID == "sp|P78347|GTF2I_HUMAN") %>%
   #filter(experimentiname== "HEKR_IEX2") %>%
   #filter(ProteinID == "sp|P45974|UBP5_HUMAN") %>%
   #filter(experiment_name== "HEMO_IEX1") 
   filter(ProteinID == "sp|Q13576|IQGA2_HUMAN") %>%
   filter(experiment_name== "HEK_SEC1") 

# See output of test
possibly_get_boundaries(tester) %>% 
  left_join(select(tester, -peak), by = "FractionID") %>%
   ggplot(aes( color = as.factor(peak), alpha = pepcount)) +
       geom_point(aes(x = Start, y = FractionID)) +
   facet_wrap(~experiment_name, scales = "free_y")

```

Peak fitting
```{r}
# About an hour
plan("multisession")

peaks <-  
  short_tidy_unique %>%

  filter(pepcount > 0) %>%
  
  #filter(experiment_name == "ARATH_IEX1") %>%
  #filter(grepl("VILI4_ARATH", ProteinID)) %>%
  split(list(.$ProteinID, .$experiment_name)) %>%
     future_map_dfr(~possibly_get_boundaries(.x), .id = "id", .progress = TRUE) %>% 
  separate(id, into = c("ProteinID", "experiment_name"), sep = "[.]")
```

```{r}

# View peaks on protein
short_tidy_unique %>% 
  inner_join(peaks)  %>%
    #filter(grepl("VILI4_ARATH", ProteinID)) %>% 
  
   filter(ProteinID == "sp|P78347|GTF2I_HUMAN") %>%
  ggplot(aes( color = as.factor(peak), alpha = pepcount)) +
       geom_point(aes(x = Start, y = FractionID)) +
  facet_wrap(~experiment_name, scales = "free_y")



peaks %>% write_csv("/project/cmcwhite/data/peptide_elutions/peaks_rms.csv")
peaks <- read_csv("/project/cmcwhite/data/peptide_elutions/peaks_rms.csv")


# Only dataset of +/- labeled proteins for testing
#short_tidy_unique_labeled_peaks <-  short_tidy_unique_labeled %>%
#     left_join(peaks,  by = c("ProteinID","experiment_name", "FractionOrder"))


# Full dataset
short_tidy_unique_peaks <-  short_tidy_unique %>%
     left_join(peaks,  by = c("ProteinID","experiment_name", "FractionOrder"))

```


### Checking peak fitting in proteins-of-interest
```{r}
  short_tidy_unique_peaks %>% 
  
   
  filter(grepl("NASP", ProteinID)) %>%

  filter(experiment_name == "HEK_SEC1") %>%

  ggplot(aes( color = as.factor(peak), alpha = pepcount)) +
       geom_point(aes(x = Start, y = FractionID)) +
  facet_wrap(~experiment_name, scales = "free_y")


short_tidy_unique_peaks %>% 
    filter(grepl("RH38_ARATH", ProteinID)) %>% 
  ggplot(aes( color = as.factor(peak), alpha = pepcount)) +
       geom_point(aes(x = Start, y = FractionOrder)) +
  facet_wrap(~experiment_name, scales = "free_y")

short_tidy_unique_peaks %>% 
    filter(grepl("MTV1_ARATH", ProteinID)) %>% 
  ggplot(aes( color = as.factor(peak), alpha = pepcount)) +
       geom_point(aes(x = Start, y = FractionID)) +
  facet_wrap(~experiment_name, scales = "free_y")



```







First classify by eye
Then classify by fit
Then classify fit peaks manually

This is applied to peptides under each fitted peak, to get maximal terminal skew.


### Filtering peaks by # of peptides
```{r}
numpeps_sel <- numpeps %>% filter(numpeps >= 9)

# Remove benzonzase experiments
# Expand peptides with observable peptides
short_tidy_unique_peaks_expanded <- short_tidy_unique_peaks %>%
  semi_join(numpeps_sel) %>% # Get rid of proteins with <9 unique peptides per an experiment
     
  select(-ExperimentID, -spec, -ID, -experiment_order) %>% # Unused columns
  select(-FractionID, -pepcount) %>% unique %>%
   mutate(pepid = paste0(Start, "-", End)) %>%
   mutate(presence = "observed") %>%
  
    group_by(ProteinID) %>%
     complete(nesting(pepid, Peptide), nesting(experiment_name, peak)) %>%
   ungroup %>%
     separate(pepid, into = c("Start", "End"), sep = "-", convert = TRUE, remove = FALSE) %>% # get start and end for filled in peptides
    mutate(presence = replace_na(presence, "missing")) # Filled in peptides get zero presence

# Get peaks that have at least 7 unique peptides
enough_cov_peaks <- short_tidy_unique_peaks %>%
   filter(!is.na(peak)) %>%
    group_by(ProteinID, peak, experiment_name) %>%
    mutate(n = n()) %>%
  ungroup %>%
  filter(n >= 7) %>%
  select(-n)

short_tidy_unique_peaks_expanded_fragments  <- short_tidy_unique_peaks_expanded %>%
  filter(!is.na(peak)) %>%
  semi_join(enough_cov_peaks, by = c("ProteinID", "peak", "experiment_name"))
  
short_tidy_unique_peaks_expanded_fragments  %>%
  write_csv("data/short_tidy_unique_peaks_expanded_fragments.csv")


short_tidy_unique_peaks_expanded_fragments <- read_csv("data/short_tidy_unique_peaks_expanded_fragments.csv")
```

# Part 2: Calculate terminal bias
Calculation observable N -> C
                           N - > Observable C
### Calibrating terminal scores
# This is done with terminal bias.R
```{r}


oneexp <- short_tidy_unique_fragments %>%
  filter(grepl("VILI4_ARATH", ProteinID)) %>% 
  filter(experiment_name == "ARATH_SEC1") %>%
  filter(peak ==4) %>%
  arrange(Start, End)

get_terminal_ranges <- function(oneexp){
    firstobs <- oneexp %>%
       filter(presence == "observed") %>%
       filter(Start == min(Start)) %>% pull(Start) %>% head(1)
    lastobs <- oneexp %>%
       filter(presence == "observed") %>%
       filter(Start == max(Start)) %>% pull(Start) %>% head(1)
          
    # observed = in this experiment, possible = seen at all across experiments / detectable
    # Look from first possible peptide to last observed peptide
    missing_n <- oneexp %>%
       filter(Start <= lastobs)
    # Look from first observed peptide to last possible peptide
    missing_c <- oneexp %>%
       filter(Start >= firstobs)
    return(list(missing_n$Start, missing_c$Start))
}

get_missing_obs_counts <- function(start, oneexp){

    df <-oneexp %>%
     group_by(ProteinID, peak, experiment_name) %>%
        mutate(fragment_end = case_when(Start < {{ start }}  ~ "nterm",
                                 Start >= {{ start }} ~ "cterm")
               ) %>% ungroup %>%
    group_by(ProteinID, peak, experiment_name, fragment_end) %>%
        dplyr::count(presence) %>% # Overwritted by seqinr
      ungroup %>%
     mutate(n= n + 1) %>%
     pivot_wider(names_from = c(presence, fragment_end), values_from = n, values_fill = 1) %>%
      mutate(bp_start = {{ start }})
    return(df)
}


get_bps <- function(oneexp){
    # Terminal bias calculation
    missing <- get_terminal_ranges(oneexp)
    missing_n <- missing[[1]]
    missing_c <- missing[[2]]

    # C terminal fragments are missing n termini
    c_fragments <- map_dfr(missing_n, ~get_missing_obs_counts(., oneexp)) %>%
      mutate(terminal = "cterm_extension")
    # N terminal fragments are missing c termini
    n_fragments <- map_dfr(missing_c, ~get_missing_obs_counts(., oneexp)) %>%
      mutate(terminal = "nterm_extension")


    fragments <- bind_rows(n_fragments, c_fragments) %>%
            mutate_at(vars(matches("[nc]term")), replace_na, 1) 
    
  
    fragments <- fragments %>%
      
      mutate(       ratio_cterm = observed_cterm/(observed_cterm + missing_cterm),
                    ratio_nterm = observed_nterm/(observed_nterm + missing_nterm)
                    ) %>% 
      #select(-breakpointleft_pepid, -breakpointright_pepid) %>%
       mutate(fc = ratio_nterm/ratio_cterm) %>%
       mutate(log2fc = log2(fc)) %>%
       mutate(abs_log2fc = abs(log2fc))

  return(fragments)
  
}


possibly_get_bps <- possibly(get_bps, otherwise = data.frame())

bps <- short_tidy_unique_peaks_expanded_fragments %>%
  
  arrange(Start, End)%>%
  split(list(.$ProteinID, .$experiment_name, .$peak)) %>%
  future_map_dfr(~possibly_get_bps(.))


bps %>% write_csv("scanning_breakpoints_all.csv")

bps <- read_csv("scanning_breakpoints_all.csv")


```

### Calculate terminal bias for each peak
```{r}

# Get which peptides were observed in each peak
obs_key <- short_tidy_unique_peaks_expanded_fragments %>%
  select(ProteinID, experiment_name, peak, Start, presence) %>% unique

# Need to get either next observed peptide past peak for C-terminal fragment 
# or previous observed pepide prior to peak for N-terminal fragment



bps_identified <- bps %>%
  group_by(ProteinID, experiment_name, peak) %>%
      filter(abs_log2fc == max(abs_log2fc))%>% data.frame() %>%

  left_join(obs_key, by = c("ProteinID", "experiment_name", "peak")) %>%
     arrange(Start) %>%
  filter(presence == "observed") %>%
  mutate(diff = Start - bp_start) %>% 
  mutate(remove_flag = case_when(log2fc > 0 & diff >0  ~ TRUE, # Only look to left of N-term frag
                                       
                           log2fc < 0 & diff < 0 ~ TRUE,
                           TRUE ~ FALSE
                            )) %>% #Only look toward C-terminal for C-term frag
  filter(remove_flag != TRUE) %>%
  filter(diff !=0) %>% # Always offset by 1 
  arrange(diff ) %>%
  group_by(ProteinID, experiment_name, peak) %>%
    mutate(terminal = case_when(log2fc > 0 ~ "nterm",
                              log2fc < 0 ~ "cterm")) %>%
   mutate(actual_terminal_peptide = case_when(log2fc > 0 & diff == max(diff) ~ Start,
                                       
                                              log2fc < 0 & diff == min(diff) ~ Start)) %>%
  filter(!is.na(actual_terminal_peptide)) %>%
  
   # Fragments must contain at least 3 peptides. Missing ends must contain at least 3 peptides
   # Hasn't been calibrated except for nterm and missing c term
     mutate(toosmallfrag=  case_when(
     terminal == "nterm" & observed_nterm <= 3 ~ "toosmall", # Actually only 2 observed
     terminal == "cterm" & observed_cterm <= 4 ~ "toosmall",# Actually only 2 observed
     terminal == "nterm" & missing_cterm <= 3 ~ "toosmall", # Actually missing 2 peptides
     terminal == "cterm" & missing_nterm < 3 ~ "toosmall",
     TRUE  ~ "fine"
     
   )) %>%
  filter(toosmallfrag == "fine") %>%
  select(-toosmallfrag) %>% 
  unique %>%
  ungroup

bps_identified %>%
  write_csv("data/bps_identified.csv")

bps_identified <- 
  read_csv("data/bps_identified.csv")

# Then go to elution_viewer.Rmd to look
```



# START of Fisher's exact section

```{r}
# Get the end of each peptide as a candidate breakpoint to scan with Fisher's exact test
breakpoints <- short_tidy_unique_peaks_expanded_fragments %>% select(ProteinID, End) %>% unique() %>% rename(midpoint = End)
```



# Create a contingency table based on each of the candidate breakpoints in the peak
```{r}
# Max size of contingency_full is num_midpoints * numpeaks * numprots * 4 
# Nterm obs, Nterm missed, Cterm obs, Cterm missed
contingency_full_step1 <- short_tidy_unique_peaks_expanded_fragments %>% 
  left_join(breakpoints) %>%
  mutate(pep_half = if_else(End <= midpoint, "firsthalf", "secondhalf")) %>%
  select(ProteinID, Peptide, experiment_name, peak, pep_half, presence, midpoint) %>%
  group_by(ProteinID, experiment_name, peak, pep_half, presence, midpoint) %>%
    summarize(n = n()) #%>%
```

```{r}

all_combinations_fixed <-  contingency_full_step1 %>%
  group_by(ProteinID, peak, midpoint, experiment_name) %>%
    complete( pep_half = c("firsthalf", "secondhalf"), presence = c("missing", "observed"), fill = list(n = 0))


all_combinations_fixed %>% write_csv("all_combinations_fixed.csv")
contingency_full <- all_combinations_fixed


all_combinations_fixed %>% select(experiment_name) %>% pull() %>% unique

```


```{r}

# Create the folder if it doesn't exist
dir.create("all_combinations_fixed", showWarnings = FALSE)

# Split the dataset by 'experiment_name' and write each to a CSV
all_combinations_fixed %>%
  group_by(experiment_name) %>%
  group_split() %>%
  purrr::walk(~write_csv(.x, file = paste0("all_combinations_fixed/", .x$experiment_name[1], ".csv")))


# Here run scripts/fisher.R input.csv output.csv

# List all output CSV files
output_files <- list.files(path = "all_combinations_fixed/", pattern = "output.*.csv", full.names = TRUE)

# Read all files and combine them
fisher_output_full <- output_files %>%
  purrr::map_dfr(read_csv)

# Save the combined result
write_csv(fisher_output_full, "fisher_output_full_combined.csv")

# At this point, in all_combinations_fixed/, we have a p.value for every potential breakpoint. 
# Now do we need to find the lowest p-value and compare it to our custom score at each potential breakpoint. 

```



```{r}
# First do peak-wise BH, then pick the minimum of that to prioritize protein
# Then for the annotation step later, can take a hand annotated peak, and calculate the same BH score and the terminal bias score

fisher_output_full_bh <- fisher_output_full %>%
  group_by(ProteinID, experiment_name, peak) %>%
     mutate(peaklevel_p_value_bh = p.adjust(p.value, method = "BH")) %>%
  ungroup()

fisher_output_summary <- fisher_output_full_bh %>% 
    group_by(ProteinID) %>%
     
     slice_min(peaklevel_p_value_bh, n = 1) %>%
  rename(min_protlevel_p_value_bh = peaklevel_p_value_bh) %>%
  ungroup() %>%
   arrange(min_protlevel_p_value_bh) %>%
  mutate(
    #p_value_bonferroni = p.adjust(min_protlevel_p_value_bh, method = "bonferroni"),
    p_value_bh = p.adjust(min_protlevel_p_value_bh, method = "BH") # Benjamini-Hochberg
  ) %>%
  ungroup() %>%
  unique()%>%
  mutate(min_p_value = min_protlevel_p_value_bh) # Not using second BH 

```



# Part 3: Curating proteins with high terminal bias or fisher's p-value
#### Select proteins with high terminal bias, then confirm with elution_viewer.Rmd
```{r}
fisher_ranking <- fisher_output_summary %>%
  arrange(p_value_bh) %>%
  rowid_to_column(var = "fisher_rank")%>%
  mutate(prioritized_fisher = case_when(-log10(p_value_bh) >= 6 ~ TRUE, TRUE ~ FALSE))


tb_ranking <- scored_labeled_all %>%
  arrange(desc(max_abs_log2fc)) %>%
  rowid_to_column(var = "tb_rank") %>%
  mutate(prioritized_tb = case_when(max_abs_log2fc >= 3 ~ TRUE, TRUE ~ FALSE))

combined_ranking <- tb_ranking %>%
  inner_join(fisher_ranking) %>%
  mutate(prioritized = case_when(prioritized_tb == TRUE & prioritized_fisher == TRUE ~ "BOTH",
                                 prioritized_tb == TRUE ~ "Terminal bias",
                                 prioritized_fisher == TRUE ~ "Fisher"))

combined_ranking %>%
  
  ggplot(aes(x = fisher_rank, y = tb_rank, color = prioritized)) +
  geom_point(alpha = 0.7) +
  scale_x_log10()

combined_ranking %>%
  left_join(labels) %>%
  ggplot(aes(x = fisher_rank, y = tb_rank, color = prioritized)) +
  geom_point(alpha = 0.7) +
  scale_x_log10() +
  facet_wrap(~label)


# Prioritized proteins based on the max abs log2fc. 
# 
priority <- bps_identified %>% 
  separate(ProteinID, into =c(NA, NA, "Entry_name"), sep = "[|]") %>% 
  
  group_by(Entry_name) %>% 
  summarize(max_abs_log2fc = max(abs_log2fc)) %>%
  ungroup %>%
  filter(max_abs_log2fc >= 3) %>%
  #filter(!Entry_name %in% x) %>% 
 # filter(grepl("HUMAN", Entry_name)) %>%
  mutate(checkecd = case_when(Entry_name %in% x ~ TRUE))  %>%
  arrange(desc(max_abs_log2fc ))

more <- priority %>%
  filter(!Entry_name %in% x) 
  
  
combined_ranking %>% filter(prioritized_fisher == TRUE) %>%
  filter(prioritized_tb == FALSE) %>%
  filter(!ProteinID %in% prioritized_tb) %>%
  filter(!ProteinID %in% fragment_data$ProteinID) %>%
  select(ProteinID) %>% unique()
  
  
combined_ranking %>%
  select(ProteinID, prioritized_fisher, prioritized_tb) %>%
  unique() %>%
  group_by(prioritized_fisher, prioritized_tb) %>%
     summarize(n = n())

### How did ICAL end up on this list?
fisher_prioritized_entries <- combined_ranking %>% filter(prioritized_fisher == TRUE) %>%
  filter(prioritized_tb == FALSE) %>% View()
  filter(!ProteinID %in% prioritized_tb) %>%
  filter(!ProteinID %in% fragment_data$ProteinID) %>%
  select(ProteinID) %>% unique() %>% 
   separate(ProteinID, into = c("tmp", "Entry", "Entry_name"), sep = "[|]") %>% pull(Entry_name)

fisher_prioritized_entries
   
for(x in fisher_prioritized_entries) {
  cat(sprintf("elut_viewer('%s', short_tidy_unique, domaindis_setup)\n", x))
}


  
for(x in fisher_prioritized_entries) {
  cat(print(x))
}


  
  
```

### Using the scores to prioritize proteins to screen visually
### The additional fisher score adds on 7 proteins with fragments
### Classified fragments from elution viewer
```{r}
data_path_tb = "/project/cmcwhite/data/peptide_elutions/fragment_categorizing_tb/"
fragment_files_tb <- dir(data_path_tb, pattern = "*csv") 

fragment_data_tb <- tibble(filename_proc = fragment_files_tb) %>%
               mutate(file_contents = map(filename_proc, ~ read_csv(file.path(data_path_tb, .)))) %>%
               unnest(cols = c(file_contents)) %>%
               mutate(from = "tb")

fragment_data_tb <- fragment_data_tb %>% select(-source, -hyperscore)
fragment_data_tb <- fragment_data_tb %>% unique() # sometimes clicked twice

data_path_fisher = "/project/cmcwhite/data/peptide_elutions/fragment_categorizing_fisher/"
fragment_files_fisher <- dir(data_path_fisher, pattern = "*csv") 

fragment_data_fisher <- tibble(filename_proc = fragment_files_fisher) %>%
               mutate(file_contents = map(filename_proc, ~ read_csv(file.path(data_path_fisher, .)))) %>%
               unnest(cols = c(file_contents)) %>%
               mutate(from = "fisher")

fragment_data_fisher <- fragment_data_fisher %>% select(-source, -hyperscore)
fragment_data_fisher <- fragment_data_fisher %>% unique() # sometimes clicked twice



fragment_data <- bind_rows(fragment_data_fisher , fragment_data_tb)
# 


fragment_data %>% select(-Start, -End, -Peptide, -FractionOrder, -FractionID, -pepcount, -peakid, -candidate, -totfracs, -experiment_order, -ExperimentID, -filename_proc, -experiment_name) %>% unique() %>% 
    group_by(from, term) %>% 
    summarise(n = n())

```

fragment_data for rescoring based on new manual peak identification
```{r}
peaks_manual <- fragment_data %>% select(ProteinID, experiment_name, FractionOrder, peakid) %>%
  unique() %>%
  rename(peak = peakid)

peaks_manual %>% write_csv("peaks_manual.csv")


short_tidy_unique_manual <- short_tidy_unique %>% filter(ProteinID %in% peaks_manual$ProteinID) %>%
  filter(experiment_name != "HEKB_SEC1") %>%
  filter(experiment_name != "SOYBN_SEC2") %>%
   filter(experiment_name != "SOYBN_SEC1") %>%
  unique()

short_tidy_unique_manual %>%
  write_csv("data/short_tidy_unique_peaks_manual.csv")

```

Now move to terminal_bias.R
Rscript scripts/terminal_bias.R data/short_tidy_unique_peaks_manual.csv peaks_manual.csv data/short_tidy_unique_peaks_expanded_fragments_tb_manual.csv

```{r}
short_tidy_unique_peaks_expanded_fragments_tb_manual_fc <- read_csv("data/short_tidy_unique_peaks_expanded_fragments_tb_manual.csv")

```




# Calculate Fisher's exact test for manually annotated fragments

```{r}
# Get the end of each peptide as a candidate breakpoint to scan with Fisher's exact test
breakpoints_manual <- short_tidy_unique_peaks_expanded_fragments_manual %>% select(ProteinID, End) %>% unique() %>% rename(midpoint = End)
```



# Create a contingency table based on each of the candidate breakpoints in the peak
```{r}
# Max size of contingency_full is num_midpoints * numpeaks * numprots * 4 
# Nterm obs, Nterm missed, Cterm obs, Cterm missed
contingency_full_step1_manual <- short_tidy_unique_peaks_expanded_fragments_manual %>% 
  left_join(breakpoints_manual) %>%
  mutate(pep_half = if_else(End <= midpoint, "firsthalf", "secondhalf")) %>%
  select(ProteinID, Peptide, experiment_name, peak, pep_half, presence, midpoint) %>%
  group_by(ProteinID, experiment_name, peak, pep_half, presence, midpoint) %>%
    summarize(n = n()) #%>%
```

```{r}

# Takes around 5 minutes
all_combinations_fixed_manual <-  contingency_full_step1_manual %>%
  group_by(ProteinID, peak, midpoint, experiment_name) %>%
    complete( pep_half = c("firsthalf", "secondhalf"), presence = c("missing", "observed"), fill = list(n = 0))


all_combinations_fixed_manual %>% write_csv("all_combinations_fixed_manual.csv")
contingency_full_manual <- all_combinations_fixed_manual
```


```{r}

# Create the folder if it doesn't exist
dir.create("all_combinations_fixed_manual", showWarnings = FALSE)

# Split the dataset by 'experiment_name' and write each to a CSV
all_combinations_fixed_manual %>%
  group_by(experiment_name) %>%
  group_split() %>%
  purrr::walk(~write_csv(.x, file = paste0("all_combinations_fixed_manual/", .x$experiment_name[1], ".csv")))


# Here run scripts/fisher.R input.csv output.csv
# cd all_combinations_fixed_manual/
#ls *csv | parallel -j 18 Rscript ../scripts/fisher.R {} output_{}

# List all output CSV files
output_files_manual <- list.files(path = "all_combinations_fixed_manual/", pattern = "output.*.csv", full.names = TRUE)

# Read all files and combine them
fisher_output_full_manual <- output_files_manual %>%
  purrr::map_dfr(read_csv)

# Save the combined result
write_csv(fisher_output_full_manual, "fisher_output_full_combined_manual.csv")

# At this point, in all_combinations_fixed_manual/, we have a p.value for every potential breakpoint in each of the manually annotated protein species. 
# But we really want only the p-value at the manually annotated breakpoint from the elut_viewer

```



```{r}
# First do peak-wise BH, then pick the minimum of that to prioritize protein
# Then for the annotation step later, can take a hand annotated peak, and calculate the same BH score and the terminal bias score

# This  time, since peakid are consistent across experiment, correct only at the peak level, not peak + experiment_name
fisher_output_full_bh_manual <- fisher_output_full_manual %>%
  group_by(ProteinID, peak) %>%
     mutate(peaklevel_p_value_bh = p.adjust(p.value, method = "BH")) %>%
  ungroup()
```


# This part not needed I think
```{r}

fisher_output_summary_manual <- fisher_output_full_bh_manual %>% 
    group_by(ProteinID) %>%
     
     slice_min(peaklevel_p_value_bh, n = 1) %>%
  rename(min_protlevel_p_value_bh = peaklevel_p_value_bh) %>%
  ungroup() %>%
   arrange(min_protlevel_p_value_bh) %>%
  mutate(
    #p_value_bonferroni = p.adjust(min_protlevel_p_value_bh, method = "bonferroni"),
    p_value_bh = p.adjust(min_protlevel_p_value_bh, method = "BH") # Benjamini-Hochberg
  ) %>%
  ungroup() %>%
  unique()%>%
  mutate(min_p_value = min_protlevel_p_value_bh) # Not using second BH 

```




Remove outlier peptides
Don't want a low confidence peptide to be the boundary
```{r}

over2_peptide_counts <- fragment_data %>% 
  group_by(Peptide, Start, End, peakid) %>%
     summarize(n = sum(pepcount)) %>%
  ungroup() %>%
  filter(n >= 2) %>%
  select(-n)

unique_peps_per_fragment <- fragment_data %>% 
  inner_join(over2_peptide_counts) %>%
  group_by(ProteinID, peakid) %>%
     summarize(distinct_peptides = n_distinct(Peptide)) %>%
  ungroup() 


fragment_data_terminal <- fragment_data %>%
  inner_join(over2_peptide_counts) %>%
  filter(!experiment_name %in% c("HEKR_IEX1", "HEKR_IEX2", "HEKB_SEC1")) %>%
  arrange(experiment_name) %>%
    group_by(ProteinID, peakid, term, seqlen) %>%
         summarize(nterm_pos = min(Start), cterm_pos =  max(End), first_pep = first(Peptide), last_pep = last(Peptide), exps= paste(unique(experiment_name), collapse = ", ")) %>%
  ungroup %>%
  select(ProteinID, peakid, term, nterm_pos, cterm_pos, seqlen, first_pep, last_pep, exps) %>%
  separate(ProteinID, into = c(NA, "Entry", "Entry_name"), sep = "[|]", remove = FALSE) %>%
   mutate(loc = case_when(term == "nterm" ~ cterm_pos, 
                          term == "cterm" ~ nterm_pos)) %>%
  mutate(in_SEC = case_when(grepl("SEC", exps) ~ TRUE)) %>%
  mutate(in_IEX = case_when(grepl("IEX", exps) ~ TRUE)) 

fragment_data_terminal %>% write_csv("/project/cmcwhite/data/fragment_data_terminal.csv")
fragment_data_terminal <- read_csv("/project/cmcwhite/data/fragment_data_terminal.csv")

```


# Isoform comparison
### Table of what parts of sequence are missing from uniprot
```{r}
uniprot_missing_human <- read_tsv("uniprot-human-missing-from-isoform.tab")
uniprot_missing_arath <- read_tsv("uniprot-arath-missing-from-isoform.tab")


uniprot_missing  <- bind_rows(uniprot_missing_human, uniprot_missing_arath)
uniprot_missing_tbl <- uniprot_missing  %>% rename(annot = `Alternative sequence`)  %>%
  filter(!is.na(annot)) %>%
  separate_rows(annot, sep= "VAR_SEQ ") %>%
  separate(annot, into = c("altseq", "annot"), sep = ";", extra = "merge") %>%
  separate_rows(`annot`, sep = ";") %>%
  filter(grepl("Missing", annot)) %>%
  mutate(isoforms = str_extract_all(annot, "isoform [a-zA-Z0-9]{1,5}")) %>%
  unnest(isoforms) %>% select(-annot) %>%
  #(altseq = str_replace(altseq, "VAR_SEQ ", "")) %>%
  separate(altseq, into = c("missing_start", "missing_end"), sep = "[.][.]") %>%
  mutate(across(starts_with("missing_"),
                as.numeric)) %>%
  filter(!is.na(missing_end)) %>%
  left_join(meta) #%>%
  #select(-Entry_name)

# Only truncations 
uniprot_truncation_variants <- uniprot_missing_tbl %>%
  unique() %>%
  group_by(Entry,  isoforms) %>%
    mutate(n = n()) %>%
  ungroup %>%
  filter(n == 1) %>% #filter(missing_start == 1 | missing_end == seqlen) %>%
  mutate(term = case_when(missing_start == 1 ~ "cterm", missing_end == seqlen ~ "nterm")) %>%
  filter(!is.na(term))  %>%
  mutate(nterm_pos = case_when(term== "nterm" ~ 1, 
                               term == "cterm" ~ missing_end + 1)) %>%
   mutate(cterm_pos = case_when(term== "nterm" ~ missing_start - 1, 
                                term == "cterm" ~ as.double(seqlen))) %>%
  select(-seqlen, -n)
  
 

```



### This is the best way to do isoform comparison so far
```{r}
candidate_isoform_matches <- fragment_data_terminal %>%

  separate(ProteinID, into = c(NA, "Entry", NA), sep = "[|]", remove = FALSE) %>%

  inner_join(uniprot_truncation_variants, by = c("Entry", "term"), suffix = c("_obs", "_iso")) %>% 
  select(-missing_start, -missing_end, -Entry) %>% 
  select(ProteinID, nterm_pos_obs, cterm_pos_obs, nterm_pos_iso, cterm_pos_iso, term,  isoforms, exps, seqlen) %>%
  filter(nterm_pos_obs >= nterm_pos_iso) %>%
    filter(cterm_pos_obs <= cterm_pos_iso) %>%
  unique() 


identified_isoforms <- candidate_isoform_matches %>% 
   mutate(obs_len = cterm_pos_obs - nterm_pos_obs) %>%
     mutate(iso_len = cterm_pos_iso - nterm_pos_iso) %>%
  mutate(ratio = obs_len / iso_len)  %>%
  arrange(desc(ratio)) %>% 
  filter(iso_len - obs_len < 500) %>% 
  filter(ratio > 0.5) %>%
  mutate(obs_span = paste0(nterm_pos_obs, "-", cterm_pos_obs )) %>%
  mutate(iso_span = paste0(nterm_pos_iso, "-", cterm_pos_iso )) %>%
  filter(!grepl("GCF", ProteinID)) %>%# Not close enought
  mutate(isoforms = case_when(isoforms == "isoform Crk" ~ "isoform Crk-I", 
                              TRUE ~ isoforms))   # Manual adjust around bad regex
identified_isoforms  %>% View


library(kableExtra)

identified_isoforms_formatted <- identified_isoforms %>% 

  select(ProteinID, Terminus = term, Observed = obs_span, Isoform = iso_span, `Isoform ID` = isoforms, `Sequence length` = seqlen, `Observed experiments` = exps) 
 
kable(identified_isoforms_formatted, "html", booktabs = T, padding = 0) %>% 
  kable_styling()
# Table 1

# , #col.names = c("virNOG ID", "In text name", "Arabidopsis Uniprot", "Arabidopsis Locus")) %>%

# BTF3_HUMAN, matches isofo rm 2, but has N terminus?
# CATC_HUMAN has additional peptide that reaches to 106 (one count), but it does have a c-terminus. could check isoform specific peptide
# WDR44_HUMAN matches isoform 3
# NLTP_HUMAN matches isoform Scp
# PUR2_HUMAN matches isoform short
# RFC4 could match isoform 
# LARP4_HUMAN matches isoform 7
# NUP98_HUMAN matches isoform 3
# AMOT_HUMAN matches isoform 2
# CRK_HUMAN matches isoform Cr


#BTF3_HUMAN, CATC_HUMA


```
UBP15 has a fragment that's a similar length to isoform3, but has a different terminal peptide. 
Combined with the c-terminal observation, likely a proteolysis


# What domains are in which fragments?
```{r}

# Get domains fully contained in fragments
fragment_domains_auto <- fragment_data_terminal %>% 
  inner_join(meta) %>%
  filter(!is.na(seqlen)) %>%

  left_join(domains_pfam) %>%
     filter(ProtBegin >= nterm_pos & ProtEnd <= cterm_pos) 

fragment_domains_auto_annot <- fragment_domains_auto %>% select(ProteinID, peakid, term, Interpro_ID, Annotation) %>%
  unique() %>%
  group_by(ProteinID, peakid, term) %>%
    summarize(Interpro_IDs= paste0(Interpro_ID, collapse = ', '),Domains= paste0(Annotation, collapse = ', ')) %>%
  ungroup 



```


Annotate the discovered fragments with terminal bias scores and p-values
everything should be already calculated, but there's a chance some were filtered out for not meeting thresholds.
Is is best to just recalculate? 
No because, lose Benjamini hochberg correction
```{r}

fisher_output_full_annotation <- fisher_output_full %>%
  select(ProteinID, nterm_pos, cterm_pos, p.value)

fisher_output_full_annotation
fragment_data_terminal
```


# Supplemental file 2, annotation of all found fragments
Now Dataset EV1
```{r}

identified_isoforms_formatted_annot <- identified_isoforms  %>%
  select(ProteinID, term, isoforms)

orthologous_stats <- fragment_data_terminal %>%
  left_join(orthologies)  %>%
  select(ID, ProteinID) %>%
  unique %>%
     group_by(ID) %>%
       summarize(nprotsingroup = n_distinct(ProteinID)) %>%
  ungroup


 fragment_data_terminal %>%
   select(ProteinID, Entry, Entry_name, peakid, nterm_pos, cterm_pos)

 fisher_output_full_manual 
 
 
# Get start and end positions of all peptides
peptide_annot <- short_tidy %>% 
  select(ProteinID, Start, End, Peptide) %>%
  unique()


# Get each peak's minimum
fisher_output_full_bh_manual_annot <- fisher_output_full_bh_manual %>%
  select(ProteinID, peakid = peak, p.value,  peaklevel_p_value_bh) %>%
  group_by(ProteinID, peakid) %>%
    summarize(min_p_value = min(p.value), min_benjh_p_value = min(peaklevel_p_value_bh))



 
short_tidy_unique_peaks_expanded_fragments_tb_manual_fc_annot <- short_tidy_unique_peaks_expanded_fragments_tb_manual_fc %>%
   select(ProteinID, peakid = peak, abs_log2fc) %>%
  group_by(ProteinID, peakid) %>%
      summarize(min_abs_log2fc = min(abs_log2fc))

fragment_data_terminal_annotated <- fragment_data_terminal %>%
  left_join(orthologies) %>%
  left_join(orthologous_stats) %>%
  left_join(unique_peps_per_fragment ) %>%
  left_join(identified_isoforms_formatted_annot) %>%
  left_join(fragment_domains_auto_annot) %>%
  left_join(short_tidy_unique_peaks_expanded_fragments_tb_manual_fc_annot ) %>%
  left_join(fisher_output_full_bh_manual_annot)


fragment_data_terminal_annotated %>%
  select(`euNOG ID` = ID, `Truncated proteins in euNOG group` = nprotsingroup, ProteinID, `ProteoformID` = peakid, Category = term, `First observed aa` = nterm_pos, `Last observed aa` = cterm_pos, `Full sequence length` = seqlen,  `In size exclusion` = in_SEC, `In ion exchange` = in_IEX, `First observed peptide`= first_pep,  `Last observed peptide`= last_pep, `Unique peptides` = distinct_peptides, `Candidate Uniprot isoform` = isoforms, `Observed experiments` = exps, `Full domains contained`= Domains, `Interpro domain IDs`= Interpro_IDs, `Proteoform minimum BH-corrected P-value` = min_benjh_p_value, `Proteoform terminal bias score` = min_abs_log2fc) %>%
  arrange(desc(`Truncated proteins in euNOG group`), `euNOG ID`, ProteinID) %>%
  write_xlsx("DatasetEV1.xlsx")




```

# Figures 


# Figure 3A: Calpastatin
```{r}

# plots as IEX then SEC
calp_hemo <-    short_tidy_unique %>%  filter(ProteinID == "sp|P20810|ICAL_HUMAN") %>% #
 
  filter(grepl("HEMO", experiment_name)) %>%
  filter(experiment_name %in% c("HEMO_SEC2", "HEMO_IEX1")) %>%
  mutate(experiment_name = fct_relevel(experiment_name,c("HEMO_SEC2", "HEMO_IEX1") ))%>%
  b_w_plot(pubtheme = FALSE)

calp_hek <-short_tidy_unique %>%  filter(ProteinID == "sp|P20810|ICAL_HUMAN") %>% #
 
  filter(grepl("HEK", experiment_name)) %>%
  filter(experiment_name %in% c("HEK_IEX1", "HEK_SEC1")) %>%
    mutate(experiment_name = fct_relevel(experiment_name,c("HEK_SEC1", "HEK_IEX1") )) %>%
  b_w_plot(pubtheme = FALSE)

calp_hemo
calp_hek


Figure_calp <- plot_grid(calp_hek, calp_hemo)

Figure_calp  %>% ggsave("figures/Figure_calp.png", ., width = 3, height = 1.5)
Figure_calp  %>%  ggsave("figures/Figure_calp.pdf", ., width = 3, height = 1.5)


```


USP15 complex for figure 7

```{r}


# HEMO_IEX1, 2, SEC1, 2
usp15 <- short_tidy_unique %>% 
  filter(grepl("UBP15", ProteinID)) %>% 
  filter(grepl("HEMO", experiment_name)) %>%
  b_w_plot()

# HEK_IEX1, 2, SEC1, 2
usp15_hek <- short_tidy_unique %>% 
  filter(grepl("UBP15", ProteinID)) %>% 
  filter(grepl("HEK_", experiment_name))  %>%
  b_w_plot() 

nqo2 <- short_tidy_unique %>% 
  filter(grepl("NQO2", ProteinID)) %>% 
  filter(grepl("HEMO", experiment_name)) %>%
  b_w_plot() 

nqo2_hek <- short_tidy_unique %>% 
  filter(grepl("NQO2", ProteinID)) %>% 
  filter(grepl("HEK_", experiment_name)) %>%
  b_w_plot() 

# Only in hemolysate
tng2 <- short_tidy_unique %>% 
  filter(grepl("HEMO", experiment_name)) %>%
  b_w_plot(pubtheme = TRUE) 


tng2_hek <- short_tidy_unique %>% 
   filter(ProteinID == "sp|Q6ICL3|TNG2_HUMAN") %>% 
  filter(grepl("HEK_", experiment_name)) %>%
  b_w_plot(pubtheme = TRUE) 


ahsp <- short_tidy_unique %>% 
  filter(ProteinID == "sp|Q9NZD4|AHSP_HUMAN") %>% # Pretty good, not in hek though
 
  filter(grepl("HEMO", experiment_name)) %>%
  b_w_plot(pubtheme = TRUE) 


ahsp_hek <- short_tidy_unique %>% 
  filter(ProteinID == "sp|Q9NZD4|AHSP_HUMAN") %>% # Pretty good, not in hek though
 
  filter(grepl("HEK_", experiment_name)) %>%
  b_w_plot(pubtheme = TRUE) 


hemoglobin <- short_tidy_unique %>%
  filter(ProteinID == "sp|P69905|HBA_HUMAN") %>%
    filter(grepl("HEMO", experiment_name)) %>%
  b_w_plot(pubtheme = TRUE) 

plot_grid( hemoglobin, usp15,ncol = 1, rel_heights = c(0.5, 1))


rfc4 <- short_tidy_unique %>% 
  filter(ProteinID == "sp|P35249|RFC4_HUMAN") %>% 
 
#  filter(grepl("HEMO", experiment_name)) %>%
  b_w_plot(pubtheme = FALSE) 



combo <- plot_grid(nqo2, tng2, ahsp, usp15, ncol = 1, rel_heights = c(0.6, 1, 0.5, 3))
combo
combo %>%
  ggsave("figures/usp_tng2_ahsp_nqo2.png", .)
combo %>%
  ggsave("figures/usp_tng2_ahsp_nqo2.pdf", .)




combo_hek <- plot_grid(nqo2_hek, usp15_hek, ncol = 1, rel_heights = c(1, 1))
combo_hek %>%
  ggsave("figures/usp_nqo2_hek.png", .)
combo_hek %>%
  ggsave("figures/usp_nqo2_hek.pdf", .)


```


# Figure 4
# Individual elution plots
```{r}
short_tidy_unique_peaks %>% 
    filter(grepl("MTV1_ARATH", ProteinID)) %>% 
  ggplot(aes( color = as.factor(peak), alpha = pepcount)) +
       geom_point(aes(x = Start, y = FractionOrder)) +
  facet_wrap(~experiment_name, scales = "free_y")

short_tidy_unique_peaks %>% 
    filter(grepl("VILI4_ARATH", ProteinID)) %>% 
  ggplot(aes( color = as.factor(peak), alpha = pepcount)) +
       geom_point(aes(x = Start, y = FractionOrder)) +
  facet_wrap(~experiment_name, scales = "free_y")

short_tidy_unique_peaks %>% 
    filter(grepl("ACT12_ARATH", ProteinID)) %>% 
  ggplot(aes( color = as.factor(peak), alpha = pepcount)) +
       geom_point(aes(x = Start, y = FractionOrder)) +
  facet_wrap(~experiment_name, scales = "free_y")


short_tidy_unique_peaks %>% 
    filter(grepl("A0A2K3DUW9_CHLRE", ProteinID)) %>% 
  ggplot(aes( color = as.factor(peak))) +
       geom_point(aes(x = Start, y = FractionOrder)) +
  facet_wrap(~experiment_name, scales = "free_y")

short_tidy_unique_peaks %>% 
    filter(ProteinID == "sp|Q13283|G3BP1_HUMAN") %>% 
  filter(experiment_name == "HEK_IEX1") %>%
  ggplot(aes( color = as.factor(peak))) +
       geom_point(aes(x = Start, y = FractionOrder)) +
  facet_wrap(~experiment_name, scales = "free_y")


quick_plot_elution <- function(ProteinID, short_tidy_unique_peaks){
  
  short_tidy_unique_peaks %>% 
    filter(grepl({{ProteinID}}, ProteinID)) %>% 
  ggplot() +
       geom_point(aes(x = Start, y = FractionOrder, alpha = pepcount, label = paste(Peptide, End))) +
  facet_wrap(~experiment_name, scales = "free_y")
  ggplotly()

  
}



```

Preprocess for simplicity in plotting elutions
```{r}
preprocess <- short_tidy_unique %>% #_labeled_peaks%>% 
  separate(ProteinID, into = c(NA, "Entry", "Entry_name"), sep = "[|]", remove = FALSE)
```

```{r}
#elut_viewer("A0A2K3D1T9_CHLRE", short_tidy_unique, domaindis_setup) # IEX and SEC SF3B1
#elut_viewer("SF3B1_HUMAN", short_tidy_unique, domaindis_setup) # IEX and SEC SF3B1
#elut_viewer("Q9FMF9_ARATH ", short_tidy_unique, domaindis_setup) # IEX and SEC SF3B1

elut_viewer("HNRPU_HUMAN", short_tidy_unique, domaindis_setup)

#K7KQN4_SOYBN

#TC159_ARATH

preprocess %>% filter(Entry_name %in% c("RPN2_HUMAN")) %>% b_w_plot()


mtv1_plot <- preprocess %>% 
  filter(Entry_name %in% c( "MTV1_ARATH")) %>%
  b_w_plot()
mtv1_plot %>% ggsave("figures/mtv1_plot.png", .,device = "png", width = 8, height = 4,  units = "in")
mtv1_plot  %>% ggsave("figures/mtv1_plot.pdf", ., device = "pdf", width = 8, height = 4, units = "in")


gtf2I_plot <- preprocess %>% 
  filter(Entry_name %in% c( "GTF2I_HUMAN")) %>%
  b_w_plot()

gtf2I_plot %>% ggsave("figures/gtf2I_plot.png", .,device = "png", width = 8, height = 4,  units = "in")

gtf2I_plot  %>% ggsave("figures/gtf2I_plot.pdf", ., device = "pdf", width = 8, height = 4, units = "in")

tc159_plot <- preprocess %>% 
  filter(Entry_name %in% c( "TC159_ARATH", "K7KQN4_SOYBN")) %>%
  b_w_plot()

tc159_plot %>% ggsave("figures/tc159_plot.png", .,device = "png", width = 4, height = 4,  units = "in")

tc159_plot  %>% ggsave("figures/tc159_plot.pdf", ., device = "pdf", width = 4, height = 4, units = "in")


#C-terminus important for interaction with EEF1a https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7139343/
EIF2A_plot <- preprocess %>% 
  filter(Entry_name %in% c( "EIF2A_HUMAN", "A0A2K3DHP9_CHLRE", "Q9CAT3_ARATH", "EF1A3_HUMAN", "EF1A2_HUMAN", "A8J4I3_CHLRE", "A8HX38_CHLRE", "A0A2K3DGR2_CHLRE", "SYMC_HUMAN", "SYMM_HUMAN")) %>%
  b_w_plot()

#A0A2K3E124? 
EIF2A_plot <- preprocess %>% 
   filter(Entry_name %in% c( "EIF2A_HUMAN", "A0A2K3DHP9_CHLRE")) %>%
            b_w_plot()#, "Q9CAT3_ARATH", "EF1A3_HUMAN", "EF1A2_HUMAN", "A8HX38_CHLRE", "A0A2K3DGR2_CHLRE",  "A0A2K3E124_CHLRE", "A0A2K3E246_CHLRE", "A0A2K3D6L9_CHLRE", "A0A2K3DD97_CHLRE", "A8HQJ0_CHLRE")) %>%

  EIF2A_plot %>% ggsave("figures/EIF2A_plot.png", .,device = "png", width = 8, height = 4,  units = "in")

EIF2A_plot  %>% ggsave("figures/EIF2A_plot.pdf", ., device = "pdf", width = 8, height = 4, units = "in")


  
#https://www.ncbi.nlm.nih.gov/pmc/articles/PMC7139343/  "A Retrospective on eIF2A—and Not the Alpha Subunit of eIF2"


preprocess %>%    filter(Entry_name %in% c("NASP_HUMAN", "CDK1_HUMAN")) %>%   b_w_plot()

catc_plot <- preprocess %>%    filter(Entry_name %in% c( "CATC_HUMAN", "RD21B_ARATH", "RDL4_ARATH", "",  "RD21A_ARATH", "A0A2K3CNP0_CHLRE", "A0A2K3CYQ1_CHLRE",  "A0A2K3DFE4_CHLRE" )) %>%   b_w_plot()
  catc_plot %>% ggsave("figures/catcplot.png", .,device = "png", width = 8, height = 4,  units = "in")
catc_plot  %>% ggsave("figures/catc_plot.pdf", ., device = "pdf", width = 8, height = 4, units = "in")

btf3_plot <- preprocess %>%    filter(Entry_name %in% c( "BTF3_HUMAN")) %>%   b_w_plot()
  btf3_plot %>% ggsave("figures/btf3_plot.png", .,device = "png", width = 8, height = 4,  units = "in")
btf3_plot  %>% ggsave("figures/btf3_plot.pdf", ., device = "pdf", width = 8, height = 4, units = "in")

sf3b1_plot <- preprocess %>%    filter(Entry_name %in% c( "SF3B1_HUMAN")) %>%   b_w_plot()
sf3b1_plot %>% ggsave("figures/sf3b1_plot.png", .,device = "png", width = 8, height = 4,  units = "in")
sf3b1_plot  %>% ggsave("figures/sf3b1_plot.pdf", ., device = "pdf", width = 8, height = 4, units = "in")


# Do this one instead of EIF5B, in humans and chlamy
elut_viewer("EIF2A_HUMAN", short_tidy_unique, domaindis_setup) # IEX and SEC SF3B1
elut_viewer("A0A2K3DHP9_CHLRE", short_tidy_unique, domaindis_setup) # IEX and SEC SF3B1



preprocess %>%
  filter(Entry_name %in% c("DDX5_HUMAN", "DDX21_HUMAN", "DDX17_HUMAN")) %>%
  b_w_plot()


preprocess %>%
  filter(Entry_name %in% c("IF2P_HUMAN", "A0A2K3E0D8_CHLRE", "F4I420_ARATH")) %>%
  b_w_plot()


```

```{r}
EIF_plots <- preprocess %>%
  filter(Entry_name %in% c("EIF3A_HUMAN", "EIF3B_HUMAN", "EIF3B_ARATH", "EIF3A_ARATH", "A0A2K3CZ51_CHLRE", "EIF3B_ARATH", "EIF3C_ARATH", "EIF3D_ARATH", "EIF3E_ARATH", "EIF3H_ARATH", "EIF3F_ARATH")) %>%
  b_w_plot() 


 EIF_plots %>% ggsave("figures/EIF_plots.png", .,device = "png", width = 8, height = 8,  units = "in")
EIF_plots %>% ggsave("figures/EIF_plots.pdf", .,device = "pdf", width = 8, height = 8,  units = "in")


elut_viewer("EIF3A_ARATH", short_tidy_unique, domaindis_setup) # IEX and SEC SF3B1
elut_viewer("EIF3A_HUMAN", short_tidy_unique, domaindis_setup) # IEX and SEC SF3B1

```


```{r}
common_prots  <- 
   c("IF2P_HUMAN", 
                           "NPM_HUMAN", 
                           "NUP98_HUMAN",
                           "ATXB2_HUMAN", 
                           "IF4G1_HUMAN",
                           "IF4G2_HUMAN",
                           "UBP2L_HUMAN",
                           "IMB1_HUMAN",
                           "G3BP1_HUMAN",
                           "DDX21_HUMAN",
                           "HNRPM_HUMAN",
                           "SRRM2_HUMAN",
                           "TCOF_HUMAN",
                           "ACINU_HUMAN",
                           "DDX3X_HUMAN",
                           "DDX6_HUMAN",
                           "DDX17_HUMAN",
                           "DDX42_HUMAN",
                           "DDX46_HUMAN",
                           "DDX50_HUMAN",
                           "DHX9_HUMAN",
                           "EIF2A_HUMAN",
                           "IF2B_HUMAN",
                           "EI2BG_HUMAN",
                           "IF4B_HUMAN",
                           "IF5_HUMAN",
                           "EIF3A_HUMAN"
                           
                           
                           )
selected_common_ids <- preprocess %>%
  filter(Entry_name %in% common_prots) %>%
  pull(ID) %>% unique

preprocess %>%
  filter(ID %in% selected_common_ids) %>%
  select(Entry_name)



fragment_data %>%
  filter(Entry_name %in% common_prots) %>%
  pull(Entry_name) %>% unique

#[1] "DDX17_HUMAN" "DDX21_HUMAN" "DDX42_HUMAN" "DDX46_HUMAN" "DDX6_HUMAN"  "DHX9_HUMAN"  "G3BP1_HUMAN" "HNRPM_HUMAN" "EIF2A_HUMAN" "IF2B_HUMAN" 
#[11] "IF2P_HUMAN"  "IF4G2_HUMAN" "NPM_HUMAN"   "NUP98_HUMAN"

```


Do cases of where fragment found, but no full length(?) detected (not just disordered parts). Check images on preprocess
```{r}

# Get frag/full lengths that contain a domain
prots_w_domains_exps <- 
  fragment_data_terminal  %>%
  separate_rows(   exps, sep = ", " ) %>%
  rename(experiment_name = exps) %>%
    left_join(fragment_domains_auto_annot)  %>%
  filter(!is.na(Domains)) %>%
  select(ProteinID, experiment_name) %>%
  unique()


```

# Figure 6A: Only termini detected
```{r}
first_domain <- domains_pfam %>%
  group_by(Entry) %>%
    mutate(firstdomain = min(ProtBegin)) %>%
  ungroup %>%
  filter(ProtBegin == firstdomain) %>%
  select(Entry, Annotation, firstdomain)


last_domain <- domains_pfam %>%
  group_by(Entry) %>%
    mutate(lastdomain = max(ProtEnd)) %>%
  ungroup %>%
  filter(ProtEnd == lastdomain) %>%
  select(Entry, Annotation, lastdomain)

fragment_data_terminal_prefirst <- fragment_data_terminal %>%
    separate_rows(   exps, sep = ", " ) %>%
  rename(experiment_name = exps) %>%
  filter(term == 'nterm') %>%
    inner_join(first_domain) %>% 
  filter(cterm_pos < firstdomain) 


fragment_data_terminal_postlast <- fragment_data_terminal %>%
    separate_rows(   exps, sep = ", " ) %>%
  rename(experiment_name = exps) %>%
  filter(term == 'cterm') %>%
    inner_join(last_domain) %>% 
  filter(nterm_pos > lastdomain)

bind_rows(fragment_data_terminal_prefirst, fragment_data_terminal_postlast) %>%
  #select(ProteinID, experiment_name) %>% 
  arrange(ProteinID) %>% 
  select(ProteinID, nterm_pos, cterm_pos, term) %>% unique %>% 
  filter(ProteinID %in% proteins_w_full$ProteinID ) %>%
select(ProteinID) %>% unique()


# 28 human proteins where only peptides from termini outside of annotated domains were detected in a fraction experiment. Detection of these peptides without inspecting the peptide coverage could result in unsusp 

# 28 human proteins
# 32 Chlamy proteins
# 11 arabidopsis proteins
bind_rows(fragment_data_terminal_prefirst, fragment_data_terminal_postlast) %>%
  #select(ProteinID, experiment_name) %>% 
  arrange(ProteinID) %>% 
  select(ProteinID, experiment_name, term) %>% unique %>% 
  anti_join(experiments_w_full) %>%
  select(ProteinID) %>% unique %>%
  filter(grepl("HUMAN", ProteinID))


# experiments_w_full not perfect. Instead of this look for peaks that start after/before domain
bind_rows(fragment_data_terminal_prefirst, fragment_data_terminal_postlast) %>%
  #select(ProteinID, experiment_name) %>% 
  arrange(ProteinID) %>% 
  select(ProteinID, nterm_pos, cterm_pos, term, experiment_name) %>% 
  #select(ProteinID, experiment_name, term) %>% unique %>% 
  anti_join(prots_w_domains_exps) %>%
  group_by(ProteinID, nterm_pos, cterm_pos, term) %>%
    summarize(solo_exps = paste0(experiment_name, collapse = ", ")) %>%
  ungroup %>% 
  filter(grepl(",", solo_exps)) %>% filter(grepl("IEX", solo_exps)) %>% filter(grepl("SEC", solo_exps)) %>%
  arrange(nterm_pos, cterm_pos)# %>%
  #filter(grepl("HUMAN", ProteinID))


# USE EIF3B as example
preprocess %>% filter(Entry_name %in% c("INT3_HUMAN", "WDR44_HUMAN", "CTR9_HUMAN", "RFC1_HUMAN",  "EIF3B_HUMAN", "RU17_HUMAN", "SRP72_HUMAN")) %>%
  b_w_plot()

```





```{r}

exp_counts <- short_tidy_unique %>%  
  group_by(ProteinID, experiment_name) %>%
  summarize(counts = sum(pepcount)) %>%
  ungroup()


other_prots <- short_tidy_unique %>%  select(ProteinID, experiment_name) %>% unique %>%
  anti_join(bps_identified) %>%
  mutate(abs_log2fc = 0) # Ones with no gaussians fitted


score_distinguisher <- bps_identified %>% 
  group_by(ProteinID ) %>%
    mutate(max_abs_log2fc = max(abs_log2fc)) %>%
  ungroup %>%
  filter(abs_log2fc == max_abs_log2fc) %>%
    left_join(exp_counts) %>%
 # select(ProteinID, experiment_name,  abs_log2fc) %>%
  left_join(fragment_data_terminal_annot) %>%
  mutate(label = case_when(is.na(label) ~ "Not detected", TRUE ~ "Detected")) 
  #(aes(x = abs_log2fc, color = label, group = label)) +
    #geom_density()
  #filter(abs_log2fc >= 3) %>%

score_distinguisher %>%
  ggplot(aes(x = abs_log2fc, color = label)) +
  geom_density()  +
  scale_color_manual(values = c("blue", "grey"))


score_distinguisher_detect <- score_distinguisher %>% filter(label == "Detected")
score_distinguisher_notdetect <- score_distinguisher %>% filter(label == "Not detected")


#ggplot()
priority_plot <- ggplot() + 
  geom_point_rast(data = score_distinguisher_notdetect, color = "grey", alpha = 0.3, aes(x = counts, y = abs_log2fc)) +
  geom_point(data = score_distinguisher_detect, color = palette[1], alpha = 0.7, aes(x = counts, y = abs_log2fc)) #+
  scale_x_log10() 
  #facet_wrap(~label)

priority_plot %>% ggsave("figures/priority_plot.png", .,device = "png", width = 2.5, height = 2.2,  units = "in")

priority_plot  %>% ggsave("figures/priority_plot.pdf", ., device = "pdf", width = 2.5, height = 2.2, units = "in")

plot_grid(priority_plot, peptide_stats_plot)

```




# Extract out fragment peaks, and do individual correlations
```{r}

extra_copd <- tibble(experiment_name = c("HEK_IEX1","HEK_IEX1","HEK_IEX2","HEK_IEX2","HEK_IEX2"), spec = c("human","human","human","human","human"),
                     ProteinID = rep("cterm-1-A-sp|P48444|COPD_HUMAN", 5), FractionID = c("HEK293T_control_IEX_47a_011217", "HEK293T_control_IEX_48a_011217", "HEK293T_ctrl_IEX_SC_45a_042217", "HEK293T_ctrl_IEX_SC_46a_042217", "HEK293T_ctrl_IEX_SC_49a_042217"), prottot = c(4,3,4,2,1))


cor_setup <- fragment_data_terminal %>%
  mutate(ProteinID = paste(term, nterm_pos, cterm_pos,ProteinID, sep = '-')) %>%

  bind_rows(short_tidy_unique) %>%
  group_by(experiment_name, spec, ProteinID, FractionID ) %>%
   summarize(prottot = sum(pepcount)) %>%
  ungroup 

total_fractions <- cor_setup %>%
  group_by(ProteinID) %>%
    summarize(nfrac_obs = n()) %>%
  ungroup()




get_fragment_cors <- function(cor_setup, total_fractions, exp_counts, minscore, score = "cor"){
  

   cor_setup_filt <- cor_setup %>%

  select(-spec, -experiment_name) %>%
  group_by(ProteinID) %>%
    mutate(total = sum(prottot)) %>%
  ungroup %>%
  filter(total > 15) %>%
  select(-total) %>%
  pivot_wider(names_from = ProteinID, values_from = prottot, values_fill = 0 ) %>%
  select(-FractionID) 


   if(score == "cors"){
      cors <- cor(cor_setup_filt, cor_setup_filt, method  = "pearson") 
   }
   if(score == "profcompare"){
      print(cor_setup_filt)
      cors <- profcompare(cor_setup_filt) 
   }
   
    fragment_cors <- cors %>% 
      as_tibble(rownames = "ID1") %>% 
      gather(ID2, r, -ID1) %>% 
      filter(r >= minscore) %>%
      arrange(desc(r)) %>%
      filter(grepl("nterm", ID1) | grepl("cterm", ID1)) %>%
      filter(ID1 != ID2) %>%
      left_join(exp_counts, by = c("ID1" = "ProteinID")) %>%
      arrange(desc(nexps)) %>%
      separate(ID1, into = c(NA, "Entry1", "Entry_name1"), sep = "[|]", remove = FALSE) %>%
      separate(ID2, into = c(NA, "Entry2", "Entry_name2"), sep = "[|]", remove = FALSE) %>% 
      left_join(total_fractions, by = c("ID1" = "ProteinID")) %>%
       left_join(total_fractions, by = c("ID2" = "ProteinID")) 
   return(fragment_cors)
 
}

cor_setup %>% filter(grepl("A8I901", ProteinID)) %>% filter(spec == "chlre")  %>% get_fragment_cors(., total_fractions, exp_counts, 0.5, "cors")

cors_chlre <- cor_setup %>% filter(spec == "chlre")  %>% get_fragment_cors(., total_fractions, exp_counts, 0.5, "cors")
cors_human <- cor_setup %>% filter(spec == "human")  %>% bind_rows(extra_copd) %>% get_fragment_cors(., total_fractions, exp_counts, 0.5, "cors")
cors_arath <- cor_setup %>% filter(spec == "arath")  %>% get_fragment_cors(., total_fractions, exp_counts, 0.5, "cors")
  
source("scripts/profcompare.R")
test_profcompare <- profcompare(cor_setup_filt) 

cors_chlre <- cor_setup %>% filter(spec == "chlre")  %>% get_fragment_cors(., total_fractions, exp_counts, 0.8, "profcompare")
cors_human_profcompare <- 
  cor_setup %>% filter(spec == "human")  %>%
    
  #  filter(ProteinID %in% c("sp|Q8N0W3|FUK_HUMAN", "cterm-3-A-sp|Q9Y4E8|UBP15_HUMAN")) %>% 
     # filter(ProteinID %in% c("cterm-2-A-sp|Q29RF7|PDS5A_HUMAN", "sp|Q99618|CDCA3_HUMAN")) %>%

  #filter(ProteinID %in% c("cterm-2-A-sp|Q13263|TIF1B_HUMAN", "cterm-2-A-sp|Q8N1G4|LRC47_HUMAN")) %>% 
    get_fragment_cors(., total_fractions, exp_counts, 0.3, "profcompare")
  
cors_human_profcompare   %>% write_csv("cors_human_profcompare.csv")
cors_human_profcompare   <- read_csv("cors_human_profcompare.csv")



cors_human_profcompare %>%
  mutate(fraction_fraction = nfrac_obs.x/nfrac_obs.y) %>%
  # Get proteins that are observed in about the same # of fractions
  filter(fraction_fraction > 0.8) %>%
  filter(fraction_fraction < 1.2) %>%
  
  arrange(desc(r)) %>%
  filter(nfrac_obs.x > 100) %>% View()
  ggplot(aes(x = nfrac_obs.x, y = r, label = paste0(ID1, "-", ID2))) +
    geom_point(alpha = 0.7) +
  scale_x_log10() 


#cors_arath <- cor_setup %>% filter(spec == "arath")  %>% get_fragment_cors(., total_fractions, exp_counts, 0.7, "profcompare")
  


uniprot_arath_annot <- read_tsv("/project/cmcwhite/data/protein_complexes_plant/accessory_files/unused/uniprot_arath_annotations.txt") %>%
  select(Entry, `Protein names`)

exp_counts <- fragment_data %>% 
  mutate(ProteinID = paste(term, peakid,candidate,ProteinID, sep = '-')) %>%
  select(ProteinID, experiment_name) %>% unique %>%
  group_by(ProteinID) %>%
    summarize(nexps = n()) %>%
  ungroup %>%
   arrange(desc(nexps))

cors_arath %>% arrange(desc(r))
preprocess %>%
  filter(Entry_name %in% c("PHOT1_ARATH",  "CIP1_ARATH", "AB14C_ARATH", "PLDD1_ARATH")) %>%
  b_w_plot()

# Good matches, unknown biology, only one experiment
preprocess %>%
  filter(Entry_name %in% c("IF4G1_ARATH",  "PER45_ARATH", "Q9LHQ4_ARATH")) %>%
  b_w_plot()

# Look at combo_score > 0.6
bind_rows(cors_human, cors_arath, cors_chlre) %>%
  rowwise() %>%
  mutate(diff = min(nfrac_obs.x,nfrac_obs.y)/  max(nfrac_obs.x,nfrac_obs.y)) %>%
  mutate(combo_score = diff * r) %>% View()
  ggplot() +
  geom_point(aes(size = nfrac_obs.x, x = diff, y = r, color = combo_score, label = paste0(Entry_name1, "-", Entry_name2)), alpha = 0.7) 
  
  # Good match, only one exp though
preprocess %>% filter(Entry_name %in% c("Q6V8K6_CHLRE", "A0A2K3DHN7_CHLRE" )) %>% b_w_plot(  
  # Good match, two experiments
  
  # Get together all the vid27 guys A0A2K3CTA4_CHLRE = vid27 chlre . Wasn't there an arabidopsis one?
  #Check that that "A0A2K3DIS4_CHLRE C-tmerinual is annotated
  preprocess %>% filter(Entry_name %in% c("A0A2K3CTA4_CHLRE","A0A2K3DIS4_CHLRE" )) %>% b_w_plot()
  
preprocess %>% filter(Entry %in% c("A8I901", "A8ID95", "A8JAD5", "A0A2K3CWJ9", "A0A2K3E2J7", "A0A2K3CTA4", "A0A2K3D5T6", "A0A2K3CPC9", "A0A2K3CNL6",  "A8J353", "A8JGV8", "P42380" )) %>% b_w_plot()

#### 
# COPD chlamy A8ID95 = pretty good cLPa3
preprocess %>%
  filter(Entry %in% c("A8I901", "A8ID95", "A8JAD5", "A0A2K3CWJ9", "A0A2K3E2J7", "A0A2K3CTA4", "A0A2K3D5T6", "A0A2K3CPC9", "A0A2K3CNL6",  "A8J353", "A8JGV8", "P42380" )) %>%
  b_w_plot()

preprocess %>%
  filter(Entry_name %in% c("COPD_HUMAN", "CHM4A_HUMAN"))%>%
  b_w_plot()

# DSC1, ECM1, SPB12, def go together in erythrocytes, not sure about fragment. Very low coverage
preprocess %>%
  filter(Entry %in% c("Q08554", "Q16610", "Q96P63")) %>%
  b_w_plot()

#nterm-2-A-sp|O65570|VILI4_ARATH
#sp|Q9LV35|AIP12_ARATH
#0.446032243
#2
#Villin-4
#Actin-interacting protein 1-2


#PUR2N sometimes with PUR6, PUR2full sometimes with PUR6N
preprocess %>%
  filter(Entry_name %in% c("PUR1_HUMAN", "PUR2_HUMAN","PUR3_HUMAN", "PUR4_HUMAN", "PUR5_HUMAN", "PUR6_HUMAN"))%>%
  b_w_plot()

preprocess %>%
  filter(Entry_name %in% c("PUR2_HUMAN", "PUR6_HUMAN"))%>%
  b_w_plot()

test_cor %>% 
  as_tibble(rownames = "ID1") %>% 
  gather(ID2, r, -ID1) %>%
  arrange(desc(r)) %>%
  filter(grepl("cterm", ID1)) %>%
  filter(ID1 != ID2) %>%
  left_join(exp_counts, by = c("ID1" = "ProteinID")) %>%
  arrange(desc(nexps))

	


cor_setup%>%
  filter(ProteinID %in% c("nterm-3-A-sp|P48634|PRC2A_HUMAN", "cterm-3-A-sp|Q5JSH3|WDR44_HUMAN")) %>%
  pivot_wider(names_from= FractionID, values_from = prottot) %>% View

cor_setup%>%
  filter(ProteinID %in% c("nterm-2-A-sp|Q9Y4E8|UBP15_HUMAN", "sp|P16083|NQO2_HUMAN")) %>%
  pivot_wider(names_from= FractionID, values_from = prottot) %>% View

cor_setup%>%
  filter(ProteinID %in% c("cterm-2-A-sp|Q29RF7|PDS5A_HUMAN", "sp|Q99618|CDCA3_HUMAN")) %>%
  pivot_wider(names_from= FractionID, values_from = prottot) %>% View



cor_setup%>%
  filter(ProteinID %in% c("internal-2-A-tr|A8I2U9|A8I2U9_CHLRE", "cterm-2-A-tr|Q6UPR1|Q6UPR1_CHLRE")) %>%
  pivot_wider(names_from= FractionID, values_from = prottot) %>% View

# Pretty good match
cor_setup%>%
  filter(ProteinID %in% c("cterm-3-A-sp|O48963|PHOT1_ARATH", "sp|F4JZY1|CIP1_ARATH", "sp|Q9LZJ5|AB14C_ARATH")) %>%
  pivot_wider(names_from= FractionID, values_from = prottot) %>% View
elut_viewer("PHOT1_ARATH", short_tidy_both, domaindis_setup)
elut_viewer("CIP1_ARATH", short_tidy_both, domaindis_setup)

# Pretty nice!
# Vid27 with Mg-protoporphyrin IX chelatase
# Interpretation?
short_tidy_both_prot %>%
  filter(ProteinID %in% c("cterm-3-A-tr|A0A2K3CTA4|A0A2K3CTA4_CHLRE", "full-1-A-tr|A8I531|A8I531_CHLRE", "tr|A0A2K3DIS4|A0A2K3DIS4_CHLRE")) %>%
  pivot_wider(names_from= FractionID, values_from = prottot) %>% View


a <- short_tidy_unique %>% 
    filter(grepl("HEK", experiment_name)) %>%
  filter(ProteinID == "sp|Q8NC51|PAIRB_HUMAN") %>% 
  b_w_plot() 
b <-  short_tidy_unique %>% 
    filter(grepl("HEK", experiment_name)) %>%
  filter(ProteinID == "sp|Q8WWM7|ATX2L_HUMAN") %>% 
  b_w_plot() 
  
c <- short_tidy_unique %>% 
  filter(grepl("HEK", experiment_name)) %>%
  filter(ProteinID == "sp|P52272|HNRPM_HUMAN") %>% 
  b_w_plot() 
  

plot_grid(a,b, c,ncol = 1)


#???
a <- short_tidy_unique %>% 
    filter(grepl("HEK", experiment_name)) %>%
  filter(ProteinID == "sp|Q8N1G4|LRC47_HUMAN") %>% 
  b_w_plot() 
b <-  short_tidy_unique %>% 
    filter(grepl("HEK", experiment_name)) %>%
  filter(ProteinID == "sp|Q13263|TIF1B_HUMAN") %>% 
  b_w_plot() 

plot_grid(a,b, ncol = 1)
cor_setup%>%
  filter(ProteinID %in% c("cterm-2-A-sp|Q8N1G4|LRC47_HUMAN", "cterm-2-A-sp|Q13263|TIF1B_HUMAN")) %>%
  pivot_wider(names_from= FractionID, values_from = prottot) %>% View


# VID27 in arabidopsis has chloroplast links
# Cloned by lianyong Cre16.g657550
# Nice match
elut_viewer("A0A2K3CTA4_CHLRE", short_tidy_both, domaindis_setup)
elut_viewer("A8I531_CHLRE", short_tidy_both, domaindis_setup)

short_tidy_both_prot %>%
  filter(ProteinID %in% c("nterm-3-A-tr|A0A2K3DPK1|A0A2K3DPK1_CHLRE", "tr|A0A2K3E2S5|A0A2K3E2S5_CHLRE")) %>%
  pivot_wider(names_from= FractionID, values_from = prottot) %>% View

elut_viewer("A0A2K3DPK1_CHLRE", short_tidy_both, domaindis_setup)
elut_viewer("A0A2K3D3A9_CHLRE", short_tidy_both, domaindis_setup)

#% in common?
# weighted detection similarity?
# same frac = TRUE
# diff frac = FALSE
# tot frac = n

short_tidy_both_prot %>%
  filter(ProteinID %in% c("nterm-2-A-sp|O65570|VILI4_ARATH", "sp|Q9LV35|AIP12_ARATH")) %>%
  pivot_wider(names_from= FractionID, values_from = prottot) %>% View

short_tidy_both_prot %>%
  filter(ProteinID %in% c("cterm-3-A-tr|F4I420|F4I420_ARATH", "sp|Q56Y85|MAP22_ARATH")) %>%
  pivot_wider(names_from= FractionID, values_from = prottot) %>% View

# Ok ish, glucosidase + alcohol dehydrogenase
short_tidy_both_prot %>%
  filter(ProteinID %in% c("cterm-2-A-sp|Q9FZE0|BGL40_ARATH", "tr|Q9LK96|Q9LK96_ARATH")) %>%
  pivot_wider(names_from= FractionID, values_from = prottot) %>% View



elut_viewer("F4I420_ARATH", short_tidy_both, domaindis_setup)
elut_viewer("MAP22_ARATH", short_tidy_both, domaindis_setup)

# EIF5B with protein that removes initial methionine? MAP22 EIF2S1 associated.
# Check in chlamydomonas
short_tidy_both_prot %>%
  filter(ProteinID %in% c("cterm-3-A-tr|F4I420|F4I420_ARATH", "sp|Q56Y85|MAP22_ARATH")) %>%
  pivot_wider(names_from= FractionID, values_from = prottot) %>% View
elut_viewer("F4I420_ARATH", short_tidy_both, domaindis_setup)
elut_viewer("MAP22_ARATH", short_tidy_both, domaindis_setup)

short_tidy_both %>% filter(ProteinID == "tr|F4I420|F4I420_ARATH") %>%
  select(ID)
short_tidy_both %>% filter(ID == "KOG1144") %>%
  filter(spec== "human") %>%
  select(ProteinID)

#tr|A0A2K3E0D8|A0A2K3E0D8_CHLRE
#tr|A8J9E6|A8J9E6_CHLRE
elut_viewer("IF2P_HUMAN", short_tidy_both, domaindis_setup)
elut_viewer("MAP2_HUMAN", short_tidy_both, domaindis_setup)
```


```{r}

complexportal <- read_tsv("outside_data/complex_portal_w_fragment_12_01_20.tsv", col_names = FALSE) %>%
   select(complex_id = X1, complex_annot = X2, entries = X5) %>%
  separate_rows(entries, sep  =  "[|]") %>%
  mutate(Entry = str_extract(entries, "^......")) %>%
  select(-entries) %>% unique()
  

#bp_protein <-  fragment_data_terminal %>% separate(ProteinID, into = c(NA, "Entry", "Entry_name"), sep = "[|]", remove = FALSE) %>% unique()
#complexportal %>%
#    inner_join(bp_protein) #%>% filter(!is.na(bpprot)) %>%
#  View()

  

fragment_data_preprocess <- fragment_data_terminal %>%
    separate(ProteinID, into = c(NA, "Entry", "Entry_name"), sep = "[|]", remove = FALSE) 

fragments_in_complexportal <- complexportal %>%  
  inner_join(fragment_data_preprocess) %>%
  select(complex_id, ProteinID, Entry) %>% unique

reject <- c()
#"sp|Q92922|SMRC1_HUMAN" 
#"sp|Q13112|CAF1B_HUMAN" No
#"sp|Q6IA86|ELP2_HUMAN" Yes? Mixed evidence 
#"sp|Q9NR30|DDX21_HUMAN" 
#"sp|P04844|RPN2_HUMAN"  
#"sp|P51610|HCFC1_HUMAN"
#"sp|P20290|BTF3_HUMAN" No
#"sp|Q08211|DHX9_HUMAN"  No
#"sp|P49792|RBP2_HUMAN"  
#"sp|P52948|NUP98_HUMAN"

complexportal  %>% filter(Entry == "P04844")

#"P61165" "Q9H0U3" "Q14165" "A0A024" "P04844" "P04843" "P61803" "P0C6T2" "Q8TCJ2"
complexportal %>% filter(complex_id == "CPX-5622") %>% pull(Entry)
# CPX-1080 CRD-mediated mRNA stability complex Q08211 Q00839

#CPX-873 Nuclear pore complex P49792 P52948 

#OST CPX
#Why was P39656 missig from complexportal?
ost <- c("P0C6T2", "P61165","Q9NRP0", "Q9H0U3", "Q14165", "A0A024", "P04844", "P04843", "P61803", "P0C6T2" ,"Q8TCJ2", "P39656")
#RPN2 lineage specific extension?

preprocess <- short_tidy_unique %>% #_labeled_peaks%>% 
  separate(ProteinID, into = c(NA, "Entry", "Entry_name"), sep = "[|]", remove = FALSE) %>%
  left_join(complexportal)

 short_tidy_unique %>% filter(grepl("F4JIM7", ProteinID))  
 
   short_tidy_unique %>% filter(grepl("RPN2_ARATH", ProteinID))  
  
   
preprocess %>% filter(Entry %in% ost) %>%
  b_w_plot()

#sp|Q92922|SMRC1_HUMAN
#sp|Q13112|CAF1B_HUMAN
#sp|Q6IA86|ELP2_HUMAN 
#sp|Q14974|IMB1_HUMAN 
#sp|Q9NR30|DDX21_HUMAN
#sp|P04844|RPN2_HUMAN 
#sp|P51610|HCFC1_HUMAN
#sp|P20290|BTF3_HUMAN 
#sp|Q00839|HNRPU_HUMAN
#sp|Q08211|DHX9_HUMAN 
#sp|P49792|RBP2_HUMAN 
#sp|P52948|NUP98_HUMAN
#sp|P60900|PSA6_HUMAN 
#sp|Q8TEQ6|GEMI5_HUMAN

```

  
```{r}
#https://www.researchgate.net/figure/Overview-of-the-structure-of-the-a-COP-CTD-e-COP-heterodimer-A-Domain-structures-of_fig1_44660041 Not seen in human

preprocess %>% filter(Entry_name %in% c("COPA1_ARATH", "COPA2_ARATH", "A0A1P8B4C2_ARATH", "COPB1_ARATH", "COPB2_ARATH", "COPE1_ARATH", "COPE2_ARATH", "COPG_ARATH","Q9XIH0_ARATH","F4IKA7_ARATH")) %>%
b_w_plot()  


# COPE, COPA chlamy
preprocess %>% filter(Entry %in% c("A8IWB5","A0A2K3CZA0")) %>%
filter(experiment_name %in% c("CHLRE_SEC1", "CHLRE_SEC2")) %>%
  b_w_plot() 


# COPA(chlre, chlamdy)
preprocess %>% filter(Entry %in% c("Q9SJT9","A0A2K3CZA0")) %>%
filter(experiment_name %in% c("CHLRE_SEC1", "CHLRE_SEC2")) %>%
  b_w_plot() 



# COPD, COPA arabidopis
preprocess %>% filter(Entry %in% c("Q9SJT9", "Q93Y22")) %>%
filter(experiment_name %in% c("ARATH_SEC1")) %>%
  b_w_plot() 

# COPA arabidopis
COPA_arath_plot <- preprocess %>% filter(Entry %in% c("Q9SJT9")) %>%
filter(experiment_name %in% c("ARATH_SEC1")) %>%
  b_w_plot()
 
COPA_arath_plot %>% ggsave("figures/COPA_arath_plot.png", .,device = "png", width =5, height = 5,  units = "in")
COPA_arath_plot %>% ggsave("figures/COPA_arath_plot.pdf", ., device = "pdf", width = 5, height = 5, units = "in")




# COPD, chlamy
preprocess %>% filter(Entry %in% c("A8I901")) %>%
  filter(experiment_name %in% c("CHLRE_IEX1", "CHLRE_SEC1")) %>%
  b_w_plot() 

# COPE, human "O14579"
preprocess %>% filter(Entry %in% c("O14579")) %>%
 # filter(experiment_name %in% c("CHLRE_IEX1", "CHLRE_SEC1")) %>%
  b_w_plot() 

# Chlamy ones
preprocess %>% filter(Entry %in% c("A8JGS8", "A0A2K3DRN1", "A8I901", "A8IM71", "A0A2K3CZA0", "A8IWB5", "A8HRR9", "A8JHN4", "A8JEP9", "A0A2K3DTG4", "A8JET7", "A0A2K3DH79")) %>%
  b_w_plot()

# Chlamy copD copE
preprocess %>% filter(Entry %in% c("A8I901",  "A8IWB5")) %>%
  b_w_plot()

# Arabidopsis
preprocess %>% filter(Entry %in% c("B3H5Z4", "Q94A40", "Q9SJT9", "A0A1P8B4C2", "F4ICX0", "F4J1E5", "F4HQE7", "A0A1I9LTK3", "A0A1P8AV73", "Q9SV21", "Q9CAA0", "Q9SV20", "Q9C827", "Q8L828", "Q93Y22", "Q9SA78", "O64748", "Q0WW26", "A0A1I9LRF5", "A0A1I9LRF7", "Q9XI32", "A0A1I9LRF9", "Q940S5", "Q84LG4", "Q8H1F4")) %>%
  b_w_plot()

preprocess %>% filter(Entry %in% c("P53621", "P35606", "P53618", "P48444", "O14579", "Q9Y678", "Q9UBF2", "P61923", "Q9P299")) %>%
  b_w_plot()

# Consisestent  in chlamy and humans that COPD has a second non-COP interaction interaction, potentially missing n-terminus. What is this from?

# In chlamy, cope also has a second interaction (A8IWB5)

# TOC159/TOC75 complex A-Domain
fig_toc <- preprocess %>% filter(Entry_name %in% c("TC159_ARATH","TC753_ARATH")) %>%
  b_w_plot() 

fig_toc  %>% ggsave("figures/fig_toc.png", ., width = 3, height = 3)
fig_toc  %>%  ggsave("figures/fig_toc.pdf", ., width = 3, height = 3)

preprocess %>% filter(Entry_name %in% c("K7KQN4_SOYBN", "I1MDC4_SOYBN") )%>%
  b_w_plot()

preprocess %>% filter(Entry_name %in% c("A0A2K3CR90_CHLRE", "A8IE32_CHLRE"))  %>% b_w_plot()


```
  
  
```{r}
 

preprocess %>%
  filter(grepl("Q9NRP0", ProteinID))
#filter(grepl("A0A2K3DH22", ProteinID)) %>%
#filter(grepl("Q93Z16", ProteinID)) %>%
 

  #filter(complex_id == "CPX-1196") %>%
#filter(complex_id == "CPX-1201") %>%
#filter(complex_id == "CPX-1194") %>%
#filter(complex_id == "CPX-1223") %>%
#filter(complex_id == "CPX-1226") %>%
#filter(complex_id == "CPX-569") %>%
#filter(complex_id == "CPX-1949") %>%
#filter(complex_id == "CPX-1099") %>%
#filter(complex_id == "CPX-5622") %>%  # OST B
#filter(complex_id == "CPX-5621") %>% # OST A
 #filter(Entry %in% ost) %>%
#filter(complex_id == "CPX-809") %>%
#filter(complex_id == "CPX-676") %>%
#filter(complex_id == "CPX-1080") %>%
#filter(complex_id == "CPX-1202") %>%
#filter(complex_id == "CPX-1204") %>%
#filter(complex_id == "CPX-1205") %>%
#filter(complex_id == "CPX-1207") %>%
#filter(complex_id =="CPX-1210") %>%
#filter(complex_id =="CPX-1211") %>%
#filter(complex_id == "CPX-1213") %>%
#filter(complex_id == "CPX-1164") %>%
#filter(complex_id == "CPX-1199") %>%
#filter(complex_id == "CPX-1206") %>%
#filter(complex_id == "CPX-1225") %>%
#filter(complex_id == "CPX-1195") %>%
#filter(complex_id == "CPX-1209") %>%
#filter(complex_id == "CPX-1212") %>%
#filter(complex_id == "CPX-1227") %>%
#filter(complex_id == "CPX-1228") %>%
#filter(complex_id == "CPX-1215") %>%
#filter(complex_id == "CPX-1216") %>%
#filter(complex_id == "CPX-1217") %>%
#filter(complex_id == "CPX-1218") %>%
#3filter(complex_id == "CPX-1222") %>%
#filter(complex_id == "CPX-1224") %>%
#filter(complex_id == "CPX-4747") %>%
#filter(complex_id == "CPX-873") %>%
#filter(complex_id == "CPX-873") %>%
#filter(complex_id == "CPX-4084") %>%
#filter(complex_id == "CPX-4223") %>%
#filter(complex_id == "CPX-4224") %>%
#filter(complex_id == "CPX-4225") %>%
#filter(complex_id == "CPX-4226") %>%
#filter(complex_id == "CPX-4203") %>%
#filter(complex_id == "CPX-4206") %>%
#filter(complex_id == "CPX-4207") %>%
#filter(complex_id == "CPX-4922") %>%
 
  b_w_plot()  
   # CPX-1080 CRD-mediated mRNA stability complex Q08211 Q00839
   # N terminal fragments together
   #filter(Entry %in% c("Q08211", "Q00839")) %>% 
   # Add nup98
   # filter(Entry %in% c("Q08211", "Q00839", "P52948")) %>% 
   # CPX-873 Nuclear pore complex P49792 P52948
 #  filter(Entry %in% c("P49792", "P52948")) %>% 
   #filter(Entry %in% c("Q9NR30", "P35659", "Q9UKV3")) %>% # 
  #filter(Entry %in% c("A0A2K3DAW8", "A0A2K3DYR4", "A8I9S6", "A0A2K3CNP4"))  %>%
  #  filter(Entry %in% c("Q9UKV3", "Q15287")) %>% 
   

  ggplot() +
       geom_point(aes(x = Start, y = FractionID ,alpha = pepcount)) +
     coord_flip() +
  scale_color_manual(values = palette) +
  scale_x_reverse() +
  background_grid() +
  facet_wrap(experiment_name ~ Entry, scales = "free_y", dir = "v")





    #left_join(bps_identified_annot) %>% 
preprocess %>%
   # CPX-1080 CRD-mediated mRNA stability complex Q08211 Q00839
   # N terminal fragments together
   #filter(Entry %in% c("Q08211", "Q00839")) %>% 
   # Add nup98
   # filter(Entry %in% c("Q08211", "Q00839", "P52948")) %>% 
   # CPX-873 Nuclear pore complex P49792 P52948
   filter(Entry %in% c("P49792", "P52948")) %>% 
   #filter(Entry %in% c("Q9NR30", "P35659", "Q9UKV3")) %>% # 
  #filter(Entry %in% c("A0A2K3DAW8", "A0A2K3DYR4", "A8I9S6", "A0A2K3CNP4"))  %>%
  #  filter(Entry %in% c("Q9UKV3", "Q15287")) %>% 
   
  ggplot() +
       geom_point(aes(x = Start, y = FractionID ,alpha = pepcount)) +
     coord_flip() +
  scale_color_manual(values = palette) +
  scale_x_reverse() +
  background_grid() +
  facet_wrap(experiment_name ~ Entry, scales = "free_y", dir = "v")

preprocess %>%
   # CPX-1080 CRD-mediated mRNA stability complex Q08211 Q00839
   # N terminal fragments together
   filter(Entry %in% c("Q08211", "Q00839")) %>% 
  ggplot() +
       geom_point(aes(x = Start, y = FractionID ,alpha = pepcount)) +
     coord_flip() +
  scale_color_manual(values = palette) +
  scale_x_reverse() +
  background_grid() +
  facet_wrap(experiment_name ~ Entry_name, scales = "free_y", dir = "v")

preprocess %>%
   # CPX-1080 CRD-mediated mRNA stability complex Q08211 Q00839
   # N terminal fragments together
    filter(Entry %in% c("Q08211", "Q00839", "P52948")) %>% 
  ggplot() +
       geom_point(aes(x = Start, y = FractionID ,alpha = pepcount)) +
     coord_flip() +
  scale_color_manual(values = palette) +
  scale_x_reverse() +
  background_grid() +
  facet_wrap(experiment_name ~ Entry_name, scales = "free_y", dir = "v")

# KEEP 
#ELP2 N terminal fragment does not co-elute with ELP3 and ELP4
preprocess %>%
   # CPX-1080 CRD-mediated mRNA stability complex Q08211 Q00839
   # N terminal fragments together
    filter(Entry %in% c("Q96EB1", "Q0PNE2" ,"Q8TE02" ,"Q6IA86" ,"O95163" ,"Q9H9T3")) %>% 
  ggplot() +
       geom_point(aes(x = Start, y = FractionID ,alpha = pepcount)) +
     coord_flip() +
  scale_color_manual(values = palette) +
  scale_x_reverse() +
  background_grid() +
  facet_wrap(experiment_name ~ Entry_name, scales = "free_y", dir = "v")



preprocess %>%
   # CPX-1080 CRD-mediated mRNA stability complex Q08211 Q00839
   # N terminal fragments together
    filter(Entry %in% c("Q96EB1", "Q0PNE2" ,"Q8TE02" ,"Q6IA86" ,"O95163" ,"Q9H9T3")) %>% 
  filter(experiment_name == "HEK_SEC2") %>% 
  ggplot() +
       geom_point(aes(x = Start, y = FractionID ,alpha = pepcount)) +
     coord_flip() +
  scale_color_manual(values = palette) +
  scale_x_reverse() +
  background_grid() +
  facet_wrap(experiment_name ~ Entry_name, scales = "free_y", dir = "v")


preprocess %>%
   # CPX-1080 CRD-mediated mRNA stability complex Q08211 Q00839
   # N terminal fragments together
    filter(Entry %in% c("Q9NRP0", "A0A024", "P04844" ,"P04843", "P61803", "P61165" ,"P0C6T2" ,"P46977")) %>% 
  ggplot() +
       geom_point(aes(x = Start, y = FractionID ,alpha = pepcount)) +
     coord_flip() +
  scale_color_manual(values = palette) +
  scale_x_reverse() +
  background_grid() +
  facet_wrap(experiment_name ~ Entry_name, scales = "free_y", dir = "v")

```



## Figure 7
#### Figure for g3bp1 interactions


```{r}

# removing all HEKR_IEX1, failed RNAse treatment
experiment_list <- c("HEK_IEX1") # , "HEKR_IEX1") 
g3bp1_solo <- short_tidy_unique %>% 
  filter(grepl("G3BP1", ProteinID)) %>% 
  filter(experiment_name %in% experiment_list) %>%
  b_w_plot() +
  theme_nothing() +
  theme(panel.background = element_rect(fill = "#FFFFC6"),
        panel.spacing = unit(0.25, "lines"),
        plot.margin = unit(margin(0.5,0.5,0.5,0.5), "lines")) 



  
g3bp1_solo %>%
  ggsave("figures/g3bp1_solo.png", ., width = 2, height = 1.5)

g3bp1_solo %>%
  ggsave("figures/g3bp1_solo.pdf", ., width = 2, height = 1.5)


droplet_prots <- tibble(ProteinID = c(
  "sp|Q13283|G3BP1_HUMAN_nterm",
"sp|Q13283|G3BP1_HUMAN_cterm",
"sp|Q99700|ATX2_HUMAN",
"sp|Q8WWM7|ATX2L_HUMAN",
#"sp|P30613|KPYR_HUMAN",	
"sp|P26196|DDX6_HUMAN",	
"sp|Q96F86|EDC3_HUMAN",	
"sp|Q14694|UBP10_HUMAN",	
"sp|Q14444|CAPR1_HUMAN"),

genenames = c("G3BP1_nterm",
           "G3BP1_cterm",
"ATAXIN-2",
"ATAXIN2-like",
#"KPYR",	
"DDX6",	
"EDC3",	
"UBP10",	
"CAPRIN1"))



# The HEK IEX RNAse are technical replicates, but we're not using the IEX RNASE experiment

g3bp1 <- short_tidy_unique %>% 
  filter(grepl("G3BP1", ProteinID)) %>% 
  filter(experiment_name %in% experiment_list) %>%
  b_w_plot() +
  #theme_nothing() +
  theme(panel.background = element_rect(fill = "#FFFFC6"),
        panel.spacing = unit(0.25, "lines"),
        plot.margin = unit(margin(0.5,0.5,0.5,0.5), "lines"))

g3bp1_terms <-short_tidy_unique %>%
  filter(ProteinID == "sp|Q13283|G3BP1_HUMAN") %>%
  mutate(terminus = case_when(Start >= 321 ~ "cterm",
                              TRUE ~ "nterm")) %>%
  mutate(ProteinID = paste0(ProteinID, "_", terminus)) %>%
  select(-terminus)

drops <- short_tidy_unique  %>%
  filter(ProteinID %in% droplet_prots$ProteinID) %>%
    bind_rows(g3bp1_terms ) %>%

  filter(experiment_name %in% experiment_list) %>%

  
  group_by(ProteinID, ExperimentID, experiment_order, experiment_name, FractionOrder) %>%
    summarize(tot_count = sum(pepcount)) %>%
  ungroup %>%
  group_by(ProteinID, experiment_name) %>%
     mutate(norm_tot_count = tot_count/max(tot_count)) %>%
  ungroup %>%
    left_join(droplet_prots, by = "ProteinID") %>%
  
  ggplot(aes(x = FractionOrder, y = fct_rev(fct_relevel(genenames, droplet_prots$genenames)), fill = norm_tot_count)) + 
    geom_tile() + 
    scale_fill_gradient(low = "#FFFFC6", high = "darkorange", na.value = '#FFFFC6' ) +
    #scale_fill_gradient(low = "yellow", high = "", na.value = "yellow")
    scale_x_continuous(limits = c(0, NA),breaks = seq(0, 1000, 30) ) +
  facet_wrap(~experiment_name) +
  theme(#panel.background = element_rect(fill = "#FFFFC6"),
        panel.spacing = unit(0.25, "lines"),
        plot.margin = unit(margin(0.5,0.5,0.5,0.5), "lines"), legend.position = "none",
        axis.title.y = element_blank(),
        axis.title.x= element_blank(),
        strip.text = element_blank(), strip.background = element_blank())#,
        #panel.border = element_rect(color = "grey50")) 
  
drops %>%
  ggsave("figures/droplet_elut.png", ., width = 2, height = 1.5)

drops %>%
  ggsave("figures/droplet_elut.pdf", ., width = 2, height = 1.5)


```

cdc73 test
```{r}
cdc73_prots <- tibble(ProteinID = c(
"sp|Q6P1J9|CDC73_HUMAN",

"sp|Q8N7H5|PAF1_HUMAN",

"sp|Q6PD62|CTR9_HUMAN",	
"sp|Q8WVC0|LEO1_HUMAN",	
"sp|Q92541|RTF1_HUMAN",	
"sp|Q9GZS3|WDR61_HUMAN"), 
genenames = c(
  "CDC73",
  "PAF1",
"CTR9",

"LEO1",	
"RTF1",	
"WDR61"))



cdc73_prots

 short_tidy_unique  %>%
  filter(ProteinID %in% cdc73_prots$ProteinID) %>%
    #bind_rows(g3bp1_terms ) %>%

  filter(experiment_name %in% c("HEK_IEX1")) %>%

  
  group_by(ProteinID, ExperimentID, experiment_order, experiment_name, FractionOrder) %>%
    summarize(tot_count = sum(pepcount)) %>%
  ungroup %>%
  group_by(ProteinID, experiment_name) %>%
     mutate(norm_tot_count = tot_count/max(tot_count)) %>%
  ungroup %>%
    left_join(cdc73_prots, by = "ProteinID") %>%
  
  ggplot(aes(x = FractionOrder, y = fct_rev(fct_relevel(genenames, cdc73_prots$genenames)), fill = norm_tot_count)) + 
    geom_tile() + 
    scale_fill_gradient(low = "#FFFFC6", high = "darkorange", na.value = '#FFFFC6' ) +
    #scale_fill_gradient(low = "yellow", high = "", na.value = "yellow")
    scale_x_continuous(limits = c(0, NA),breaks = seq(0, 1000, 30) ) +
  facet_wrap(~experiment_name) +
  theme(#panel.background = element_rect(fill = "#FFFFC6"),
        panel.spacing = unit(0.25, "lines"),
        plot.margin = unit(margin(0.5,0.5,0.5,0.5), "lines"), legend.position = "none",
        axis.title.y = element_blank(),
        axis.title.x= element_blank(),
        strip.text = element_blank(), strip.background = element_blank())#,
        #panel.border = element_rect(color = "grey50")) 


```






## Good to here



### Modify annotation for later plotting
```{r}
domaindis_annot <- domaindis_setup %>%
   mutate(Annotation = case_when(Annotation == "" & set == "Disordered" ~ "Disordered",
                                 Annotation == "" & set == "Transmembrane" ~ "Transmembrane",
                                TRUE ~ Annotation)) %>%
  mutate(domain_id = paste(ProtBegin, ProtEnd, Annotation, sep = "_")) %>%
  select(Entry, ProtBegin, ProtEnd, Annotation, domain_id, seqlen ) 
```





```{r}

bps_identified  %>%
 filter(grepl("A0A2K3D371_CHLRE", ProteinID))  %>%
  View()

bps_identified %>%
   filter(grepl("EMAL4_HUMAN", ProteinID))  %>% # Disordered on c term
 # filter(experiment_name == "HEK_SEC1") %>%
  View()
bps_identified_annot<- bps_identified %>%
  arrange(desc(abs(fc))) %>%
  select(ProteinID, peak, experiment_name, actual_terminal_peptide, fc, log2fc, abs_log2fc, terminal)


short_sel_peaks %>%
  filter(grepl("AFAD_HUMAN", ProteinID)) %>% 
  inner_join(bps_identified_annot) %>% 
  group_by(ProteinID, peak, experiment_name) %>%
  mutate(in_frag = case_when(
      terminal == "nterm" & Start <= actual_terminal_peptide ~ "In fragment",
      terminal == "cterm" & Start >= actual_terminal_peptide ~ "In fragment",
      TRUE ~ "Not in fragment"
  )) %>% 

  ggplot() +
       geom_point(aes(x = Start, y = FractionID, color = as.factor(peak), alpha = pepcount)) +
       geom_vline(aes(xintercept = actual_terminal_peptide, alpha = abs_log2fc, shape = in_frag, color = as.factor(peak))) + 
  facet_wrap(experiment_name ~ peak, scales = "free_y")


```

Checking
```{r}

bps %>%
 filter(grepl("VILI4_ARATH", ProteinID)) %>% 
  ggplot(aes(x = bp_start, y = fc)) +
  geom_point()+
  facet_wrap(experiment_name ~ peak + terminal)


bps_identified %>%
 filter(grepl("VILI4_ARATH", ProteinID))




long_sel_peaks %>% 
    #filter(grepl("VILI4_ARATH", ProteinID)) %>% 
  #  filter(grepl("DNMT1_HUMAN", ProteinID)) %>%
    #filter(grepl("TPR_HUMAN", ProteinID)) %>%
    #filter(grepl("A0A2K3D5H8", ProteinID)) %>%
    # filter(grepl("AFAD_HUMAN", ProteinID)) %>%
  #filter(grepl("EMAL4_HUMAN", ProteinID)) %>%
 # filter(grepl("Q9S841", ProteinID)) %>%
   #filter(grepl("HNRPU_HUMAN", ProteinID)) %>%
     #filter(grepl("A0A2K3DD86", ProteinID)) %>%
     #filter(grepl("ERLN1_HUMAN", ProteinID)) %>%  
  
  #filter(grepl("NISCH_HUMAN", ProteinID)) %>% # N term corresponds to splice variants.C term doesn't
   # filter(grepl("G3BP1_HUMAN", ProteinID)) %>%
    #       filter(grepl("A0A2K3D371_CHLRE", ProteinID)) %>%
 
  
    left_join(bps_identified_annot) %>% 
  
   #filter(grepl("Y4554_ARATH", ProteinID)) %>% 
  ggplot() +
       geom_point(aes(x = Start, y = FractionID, color = as.factor(peak), alpha = pepcount)) +
       geom_vline(aes(xintercept = actual_terminal_peptide, alpha = abs_log2fc, color = as.factor(peak))) + 
  facet_wrap(~experiment_name, scales = "free_y")

# This is double checked
# If it's negative ( c terminal fragment ), the start is the next observed peptide toward C
# If it's positive ( n terminal fragment ), the start is the next observed peptide toward N
 
```

Do precision recall of max value per experiment

Remove if fragment doesn't contain at least three peptides = reasonable

KEEP Figure 2B



# Figure 2B

```{r}

scored_labeled_all <- bps_identified %>%    


  group_by(ProteinID, experiment_name) %>%
   filter(abs_log2fc == max(abs_log2fc)) %>%
   rename(max_abs_log2fc=abs_log2fc) %>%
  select(ProteinID, experiment_name, actual_terminal_peptide, max_abs_log2fc, terminal) %>%
    
  
    #summarize(max_abs_log2fc = max(abs_log2fc)) %>%
  #ungroup %>%
  arrange(desc(max_abs_log2fc)) %>%

    filter(experiment_name != "HEKB_SEC1")  # Add peak to show unscored 

# Where are the NAs coming from? 
# They are coming from labeled proteins that don't pass the score thresholds (number of peptides)

scored_labeled <- scored_labeled_all %>%
  full_join(labels, multiple = "all") %>%
  #full_join(alt_labels, multiple = "all") %>%
  filter(!is.na(label))  %>%
  mutate(max_abs_log2fc = replace_na(max_abs_log2fc, -1))

#scored_labeled %>%
#  filter(label == "pos" & set == "test") %>%
#  filter(max_abs_log2fc == -1)

# Didn't pass thresholds
too_few_peptides <- scored_labeled %>% filter(max_abs_log2fc == -1) %>%
  select(ProteinID, experiment_name)  %>%
    filter(experiment_name != "HEKB_SEC1")  %>%
  mutate("toofew" = TRUE)
  


scored_labeled_formatted <- scored_labeled %>%
    filter(experiment_name != "HEKB_SEC1") 

  
  #mutate(custom_label = case_when(label == "neg" & set == "train" ~ "Negative train",
  #                         label == "pos" & set == "train" ~ "Positive train",
  #                         label == "neg" & set == "val" ~ "Negative validate",
  #                         label == "pos" & set == "val" ~ "Positive validate"
  #                         
  #                         )) %>%
  
 
  mutate(custom_label = case_when(label=="pos" ~ "Hand selected truncated proteins",
                                  label == "neg" ~ "Random proteins")) %>%
   filter(!is.na(custom_label))

theme_set(theme_cowplot_consistent_text(font_size = 8))
# With abundance matching
score_dist_plot <- scored_labeled_formatted %>%
  #mutate(custom_label = fct_relevel(custom_label, c("Positive train", "Positive validate", "Negative train","Negative validate" ))) %>%
  ggplot(aes(x = max_abs_log2fc, fill = custom_label)) + 
  geom_histogram(binwidth = 0.1) + 
  background_grid(major = ("x")) +
  scale_fill_manual(values= c(palette[2], "grey")) +
  #xlab("Proteoform score") + 
 # xlab("Max absolute log2 Fold Change\n(N vs. C terminal observed/observable peptides") +
  xlab("Max terminal bias") +
  ylab("# of proteins") +
  theme(legend.position = "none", strip.background = element_blank()) +
  facet_wrap(~custom_label,scales= "free_y", ncol = 1)

score_dist_plot
scored_labeled_formatted


score_dist_plot %>% ggsave("figures/score_dist_plot.png", .,device = "png", width = 2.5, height = 2.5,  units = "in")
score_dist_plot %>% ggsave("figures/score_dist_plot.pdf", .,device = "pdf", width = 2.5, height = 2.5,  units = "in")

```


```{r}
library(broom)
result <- scored_labeled_formatted %>% 
  ## Filter data to only include groups A and B
  #filter(group %in% c("A", "B")) %>%
  # Conduct a t-test
  mutate(case_when(max_abs_log2fc == -1 ~ 0, TRUE ~ max_abs_log2fc )) %>%
  summarise(t_test = list(t.test(max_abs_log2fc ~ custom_label, data = .))) %>% 
  # Use broom's tidy() to get a tidy data frame of the results
  mutate(t_test = map(t_test, tidy)) %>%
  unnest(t_test)

# The p-value can be found in the column named "p.value"
print(result$p.value)

```


Create abundance matched sample
```{r}

# Required Libraries
library(tidyverse)

# Assuming your data frame (df) has a numeric variable (score), a grouping variable (class) 
# with levels "control" and "treatment", and an ID variable (id)
set.seed(42)  # for reproducibility
df <- data.frame(id = 1:1000,
                 class = rep(c("control", "treatment"), each = 500),
                 score = c(rnorm(500, mean = 2, sd = 1), rnorm(500, mean = 3, sd = 1))) %>% as_tibble()

# Calculate quantiles for treatment group
df <- df %>%
  mutate(quantile = ifelse(class == "treatment", ntile(score, 100), NA))

# Match controls to treatment based on quantiles
df_controls <- df %>%
  filter(class == "control") %>%
  mutate(quantile = ntile(score, 100))

df_matched <- df %>%
  filter(class == "treatment") %>%
  left_join(df_controls, by = "quantile", suffix = c("_treatment", "_control"))

# Check matched samples
head(df_matched)


```

```{r}

# Required Libraries
# Assuming your data frame (df) has a numeric variable (score) 
# and a grouping variable (class) with two levels "A" and "B"
library(ggrepel)

# max abs log2f is by protein, not by experiment
df <- scored_labeled_formatted %>% rename(score = max_abs_log2fc, class = custom_label) %>%
  select(ProteinID,  label, class, score) %>%
  unique()
# Count the number of scores greater than each value in each class


tpr <- data.frame(score2 = -1:5) %>%
  mutate(
    num_in_B = map_dbl(score2, ~sum(df$class[df$score >= .] == "Hand selected truncated proteins")), #/  sum(df$score >= .)),
    num_in_A = map_dbl(score2, ~sum(df$class[df$score >= .] == "Random proteins")),
    
      prop_in_B = map_dbl(score2, ~sum(df$class[df$score >= .] == "Hand selected truncated proteins")/  sum(df$score >= .)),
    prop_in_A = map_dbl(score2, ~sum(df$class[df$score >= .] == "Random proteins")/  sum(df$score >= .))
    #/  sum(df$score >= .)), / sum(df$score >= .))
  )

fig2_addition <- tpr %>% as_tibble() %>%
  mutate(score_label = case_when(score2 == "-1" ~ "All", 
                                 TRUE ~ paste0(">", score2 ))) %>%
  mutate(recall = num_in_B / 275) %>%
  select(score2, prop_in_B, recall, score_label) %>% 
  ggplot(aes(y = prop_in_B, x = recall, label = score_label)) + 
  geom_line(color = "grey60") + 
  geom_point(color= "grey60") + 
  geom_text_repel(force_pull = 10, size = 2.5, color = "grey20") + ylab("Precision") + xlab("Recall")

fig2_addition
fig2_addition %>% ggsave("figures/fig2_addition.png", .,device = "png", width = 2, height = 2,  units = "in")
fig2_addition %>% ggsave("figures/fig2_addition.pdf", .,device = "pdf", width = 2, height = 2,  units = "in")


#Compared to 1000 random proteins, above a max terminal bias score of 1, the true positive rate is  73% with a recall of (96.6%), above a score of 2,  the true positive rate is 91.5% with recall of (80.8%), above a score of 4, the true positive rate is  100% with a recall of 18%.

#At a score of 2, the true positive rate is 90%. 
#At a score of above 4, the true positive rate is 100%. 

#sum(tpr$prop_in_A)

fig2_addition_fdr <- tpr %>%
  ggplot(aes(x= score, y = prop_in_B)) +
   #geom_hline(yintercept= 0.95, linetype = "dashed", color = "grey50") + 
  geom_vline(xintercept= 3, linetype = "dashed", color = "grey50") +
    geom_line(color = "grey60") + 
  geom_point(color= "grey60") + 
  ylab("1 - FDR") +
  annotate("text", x = 4, y = 0.8, label = "5% FDR", color = "grey50", size = 2) +
  xlab("Max terminal bias")
fig2_addition_fdr 

  #geom_hline(yintercept = 0.95, linetype = "dashed")

# Calculate the proportion of scores in class B that are greater than each value
proportions <- counts %>%
  mutate(across(score:(score+5), ~.[class == "B"] / sum(.[class == "B"])))

# Calculate the log likelihood
log_likelihoods <- proportions %>%
  mutate(across(score:(score+5), log))

print(log_likelihoods)



```



# Figure 2C
```{r}
positives_annot <- positives %>%
  select(ProteinID, experiment_name) %>%
  unique() %>%
  mutate(label = "Hand selected")

bps_screened <- bps_identified %>%
  left_join(fisher_output_identified) %>%
  filter(abs_log2fc >=3 | p.value <= 0.001)
  

fragment_data_terminal_expand <-
  fragment_data_terminal %>%
  separate_rows(exps, sep = ", ") %>%
  rename(experiment_name = exps)

fragment_data_score <- fragment_data_terminal_expand  %>%
  filter(term %in% c("cterm", "nterm")) %>%
  select(ProteinID, experiment_name) %>%
  inner_join(bps_screened) %>%
  anti_join(positives_annot) %>%
  mutate(label = "Found with score") 

#fragment_data_score %>% View()

fragment_data_eye <- fragment_data_terminal_expand  %>%
  filter(term %in% c("cterm", "nterm")) %>%
  select(ProteinID, experiment_name) %>%
  anti_join(bps_screened)%>%
  anti_join(positives_annot)%>%
  mutate(label = "Final check")


fragment_data_terminal_annot <- 
  bind_rows(
    positives_annot,
    fragment_data_score,
    fragment_data_eye
  )

```

```{r}

mini_annot <- fragment_data_terminal_annot %>%
  select(ProteinID, experiment_name, label, peak) %>% unique()

# Fisher output only for human proteins?
# Should be ~45 "Found with score"
fisher_output_identified <- fisher_output_full  %>%
  group_by(ProteinID, experiment_name, peak) %>%
      filter(p.value == min(p.value)) # %>% data.frame()

library(scales)
reverse_log_trans <- function() {
  trans_new(
    name = 'reverse_log',
    transform = function(x) -log10(x),
    inverse = function(x) 10^(-x)
  )
}

fisher_output_identified  %>%
  inner_join(bps_identified, by = c("ProteinID", "peak", "experiment_name")) %>% arrange(p.value) %>%
  #View()

  #(10000) %>%
  left_join(mini_annot) %>% 
  ggplot(aes(x = abs_log2fc, y = p.value, label = ProteinID)) +
  geom_point() + 
  scale_y_continuous(trans = reverse_log_trans()) +
  #scale_y_log10() +
  #scale_y_reverse() +
  geom_vline(xintercept = 2.5, linetype = "dashed", color = "grey80") + 
  facet_wrap(~label, nrow =1 )

fisher_output_sel %>%
  inner_join(fisher_output_sel, by = c("ProteinID", "experiment_name", "peak")) %>%
  #sample_n(1000) %>%
     ggplot(aes(x = abs_log2fc, y = p.value)) + 
     geom_point() + 
  scale_y_log10()

```


```{r}




num_unique <-short_tidy_unique %>%
  select(ProteinID, Peptide, experiment_name) %>%
  unique %>%
  group_by(ProteinID, experiment_name) %>%
    summarise(num_unique = n()) %>%
  ungroup()

peptide_stats <- short_tidy_unique %>%

  group_by(ProteinID, experiment_name)  %>%
    summarize(num_counts = sum(pepcount)) %>% 
  ungroup() %>%
  inner_join(num_unique) %>%
  filter(num_counts > 2) %>%
  left_join(fragment_data_terminal_annot, by = c("ProteinID", "experiment_name")) %>%
  mutate(label = replace_na(label, "non_fragment"))


xxx

peptide_stats_plot <-   peptide_stats %>% filter(label != "non_fragment") %>%
    mutate(label = factor(label, levels = c("Hand selected", "Final check", "Found with score"))) %>%


ggplot( aes(x = num_unique, y = num_counts, color = label, group = label)) + #, label = paste0(ProteinID, experiment_name))) + 
  #geom_point_rast(data = peptide_stats, aes(x = num_unique, y = num_counts), color = "grey80", alpha = 0.2) +

   geom_point(alpha = 1, size = 0.5  ) +   
  #geom_density_2d(alpha = 1, size = 0.5, n= 128, bins = 3) +
  
   # scale_color_manual(values = c(palette[2], palette[1], palette[0]), name = "Fragment")  +
  scale_color_manual(values = c("Hand selected"= "#0072B2", "Found with score" = "#ff6200" , "Final check" = "#ffb001" ), name = "Fragment") +
  
  # new_scale_color() +
    #geom_density_2d(data = peptide_stats, aes(x = num_unique, y = num_counts, color = as.factor(..level..)), alpha = 0.8, n = 128, bins = 3, contour_var = "density", ) +
  #scale_color_brewer(palette = "Greys", guide = "none") + 
  #scale_color_manual(values = c(palette[1], "#ff6200" , "#ffb001" ), name = "Fragment") +
  #scale_color_manual(values = c("grey70", "grey50", "grey30"), guide = "none") +
  scale_y_log10() +
  scale_x_log10() +
  ylab("# Peptide spectral matches") +
  xlab("# Unique peptides") +
  guides(color = guide_legend(nrow = 2, title.position = "top")) +
  theme(legend.title = element_blank(), legend.position = "bottom")


peptide_stats_plot

#library(ggnewscale)
#peptide_stats_plot <-   ggplot() + 
#  #geom_point_rast(data = peptide_stats, aes(x = num_unique, y = num_counts), color = "grey80", alpha = 0.2) +#

#   geom_point(data = filter(peptide_stats, label != "non_fragment"),  
#                aes(x = num_unique, y = num_counts, color = fct_rev(label),  label = paste0(ProteinID, experiment_name)), alpha = 1, size = 0.5  ) +   
#  geom_density_2d(data = filter(peptide_stats, label != "non_fragment"),  
#                aes(x = num_unique, y = num_counts, color = fct_rev(label),  label = paste0(ProteinID, #experiment_name)), alpha = 1, size = 0.5, bins = 1 ) +
#  
#   # scale_color_manual(values = c(palette[2], palette[1], palette[0]), name = "Fragment")  +
#  scale_color_manual(values = c(palette[1], "#ff6200" , "#ffb001" ), name = "Fragment") +
#  
#  # new_scale_color() +
#    #geom_density_2d(data = peptide_stats, aes(x = num_unique, y = num_counts, color = as.factor(..level..)), #alpha = 0.8, n = 128, bins = 3, contour_var = "density", ) +
#  #scale_color_brewer(palette = "Greys", guide = "none") + 
#  #scale_color_manual(values = c(palette[1], "#ff6200" , "#ffb001" ), name = "Fragment") +
#  #scale_color_manual(values = c("grey70", "grey50", "grey30"), guide = "none") +
#  scale_y_log10() +
#  scale_x_log10() +
#  ylab("# Peptide spectral matches") +
#  xlab("# Unique peptides") +
#  guides(color = guide_legend(nrow = 2, title.position = "top")) +
#  theme(legend.title = element_blank(), legend.position = "bottom")#

peptide_stats_plot

peptide_stats_plot %>% ggsave("figures/peptide_stats_plot.png", .,device = "png", width = 3.5, height = 2.2,  units = "in")

peptide_stats_plot  %>% ggsave("figures/peptide_stats_plot.pdf", ., device = "pdf", width = 3.5, height = 2.2, units = "in")


```



Observation of these disordered termini generally requires specific terminal tags, as they tend to not 
be recognized by antibodies (IF4B, fibrocystin). 

# Figure 2D
```{r}
short_tidy_unique %>% select(experiment_name) %>%
  unique

exp_totals <- short_tidy_unique%>%
  group_by(experiment_name) %>%
    summarize(exp_totpepcounts = sum(pepcount), exp_prots = n_distinct(ProteinID))

exp_peptides <- short_tidy_unique %>%
  select(ExperimentID, experiment_name, Peptide, spec) %>%
  unique() %>%
  group_by(ExperimentID, experiment_name, spec) %>%
    summarize(exp_totpeptides = n_distinct(Peptide)) %>% ungroup

tot_positives <- 
  fragment_data_terminal_annot %>%
  unique %>%
  group_by(experiment_name) %>%
    summarize(numpos = n())

stats <- 
  tot_positives %>%
   full_join(exp_peptides, by = "experiment_name") %>%
   full_join(exp_totals, by = "experiment_name") %>%
  arrange(spec, numpos) 

stats

#palette_OkabeIto_black <- c("#E69F00", "#56B4E9", "#009E73", "#F0E442", "#0072B2", "#D55E00", "#CC79A7", "#000000")
#colorblocks <- c("#FF0000","#0072B2","#E69F00","#009E24", "#979797","#5530AA", "#111111")
#colors_key <- c("Red", "Blue", "Orange","Green",  "Grey", "Purple","Black") 
#palette <- rep(colorblocks, 20)
library(ggrepel)
library(ggtext)

stats$numpos[is.na(stats$numpos)] <- 0

stats_plot <- stats %>%
  filter(!grepl("HEKU", experiment_name)) %>%
    filter(!grepl("HEKB", experiment_name)) %>%
    filter(!grepl("MAIZE", experiment_name)) %>% 
    filter(!grepl("SOYBN", experiment_name)) %>%

mutate(Sample = case_when(
       grepl("HEK", experiment_name) ~ "HEK",
       grepl("HEMO", experiment_name) ~ "Hemolysate",
       grepl("CHLRE", experiment_name) ~ "*Chlamydomonas*",
       grepl("ARATH", experiment_name) ~ "*Arabidopsis*"
    
  )) %>%
  mutate(Sample = fct_rev(fct_relevel(Sample, c("*Chlamydomonas*",  "*Arabidopsis*","HEK", "Hemolysate")))) %>%
  mutate(labels = case_when(numpos > 25 ~ experiment_name)) %>%
  
  ggplot(aes( x = exp_totpepcounts, y = numpos, color = Sample, shape = Sample, label = Sample )) + 
  geom_point(alpha = 0.9) + 
 # geom_text_repel(size = 3)+ 
  scale_shape_manual(values = c("*Arabidopsis*" = 15, 
                                "*Chlamydomonas*" = 16, 
                                "HEK" = 17, 
                                "Hemolysate" =3)) +
  
  scale_color_manual(values = c("*Arabidopsis*" = palette[3], 
                                "*Chlamydomonas*" = palette[1], 
                                "HEK" = palette[2], 
                                "Hemolysate" =palette[4])) +
  scale_x_continuous(labels = scales::comma, breaks = c(400000,800000)) +
  guides(color = guide_legend(nrow = 2, title.position = "top"), shape  = guide_legend(nrow = 2, title.position = "top")) +
  theme(legend.title = element_blank(), legend.position = "bottom") +
  xlab("Total PSMs") + 
  ylab("# Proteins with truncation proteoforms") +
  theme(legend.text = element_markdown())

stats_plot

stats_plot %>% ggsave("figures/stats_plot.png", .,device = "png", width = 3, height = 2,  units = "in")
stats_plot %>% ggsave("figures/stats_plot.pdf", ., device = "pdf", width = 3, height = 2, units = "in")

# As peptide detection gets deeper, more proteoforms will be discoverable by this method


```

#Supplemental file 1
```{r}
library(writexl)
# Supplemental File 1
stats_formatted <- stats %>%
  filter(experiment_name != "HEKB_SEC1") %>%
  select(-numpos) %>%
  select(Species = spec, Name = experiment_name, ID = ExperimentID, Proteins = exp_prots, Peptides = exp_totpeptides, PSMs = exp_totpepcounts) %>%
  mutate(Species = case_when(Species == "human" ~ "Homo sapiens",
                             Species == "arath" ~ "Arabidopsis thaliana",
                             Species == "chlre" ~ "Chlamydomonas reinhardtii"))

stats_formatted %>%
  
  write_xlsx("Supplemental_File1.xlsx")
# Table 1

```


Figure 2 revision
```{r}
#left <- plot_grid(NULL, peptide_stats_plot + theme(legend.position = "bottom"), ncol= 1, labels = c("A", "C"), label_size = 8, rel_heights = c(1,0.9))
#right <- plot_grid(score_dist_plot, stats_plot + theme(legend.position = "bottom"), ncol = 1, labels = c("B", "D"), label_size = 8, align = "v", axis = "l", rel_heights = c(1,0.9))

topright <- plot_grid(fig2_addition, fig2_addition_fdr, labels = c("C", NULL), ncol = 1, label_size = 8)
topright

top_row <- plot_grid(NULL, new_Figure2b, topright, labels  = c("A", "B"), nrow = 1, label_size = 8)
top_row

#tlbr
bottom_row <- plot_grid( NULL,
                         peptide_stats_plot + theme(plot.margin = margin(t = 5, r = 30, b = 10, l = 20)),
                         stats_plot + theme(plot.margin = margin(t = 5, r = 50, b = 10, l = 20)),
                         nrow = 1, labels = c("", "D", "E"), label_size = 8, rel_widths =c (0.2,1,1))
bottom_row


figure2_generated_rev <- plot_grid(top_row, bottom_row, ncol = 1, rel_heights = c(1, 0.7))
figure2_generated_rev

figure2_generated_rev %>% ggsave("figures/figure2_generated_rev.png", .,device = "png", width = 7, height = 7,  units = "in")
figure2_generated_rev  %>% ggsave("figures/figure2_generated_rev.pdf", ., device = "pdf", width = 7, height = 7, units = "in")



```

Figure 2 pre revision 2
```{r}
left <- plot_grid(NULL, peptide_stats_plot + theme(legend.position = "bottom"), ncol= 1, labels = c("A", "C"), label_size = 8, rel_heights = c(1,0.9))
right <- plot_grid(score_dist_plot, stats_plot + theme(legend.position = "bottom"), ncol = 1, labels = c("B", "D"), label_size = 8, align = "v", axis = "l", rel_heights = c(1,0.9))

figure2_generated <- plot_grid(left, NULL, right, ncol = 3, rel_widths = c(1,0.2,1))

figure2_generated %>% ggsave("figures/figure2_generated.png", .,device = "png", width = 5, height = 5,  units = "in")
figure2_generated  %>% ggsave("figures/figure2_generated.pdf", ., device = "pdf", width = 5, height = 5, units = "in")



```



```{r}


library(caret)
make_pr_curve <- function( cutoff, for_pr,scorename =score){
  
    for_pr_classed <- for_pr %>%
      mutate(pred = case_when({{ scorename}} >= cutoff ~ 1,
                              {{ scorename}} < cutoff ~ -1)) %>%
      mutate(pred = as.factor(pred)) %>%
      mutate(num_label = as.factor(num_label))
      
     y <- for_pr_classed$num_label # factor of positive / negative cases
     predictions <- for_pr_classed$pred # factor of predictions  

     precision <- posPredValue(predictions, y, positive="1")
     recall <- sensitivity(predictions, y, positive="1")
     
     
     pr <- bind_cols(precision = precision, recall = recall) %>%
       mutate(cutoff = {{ cutoff}})
     return(pr)
}

for_pr <- scored_labeled %>%
  
  left_join(labels) %>%
  mutate(id = paste0(ProteinID, "-", experiment_name)) %>%
  rename(score= max_abs_log2fc) %>%
  select(id, label, set, score) %>%
  mutate(num_label = case_when(
    label == "pos" ~ 1,
    label == "neg" ~ -1
  )) 

for_pr_train <- for_pr %>%
  filter(set == "train")
for_pr_validate <- for_pr %>%
  filter(set == "val")

increments <- seq(0, max(for_pr$score), 0.1)
pr_train <- map_dfr(increments, ~make_pr_curve(., for_pr_train)) %>%
  mutate(set = "train")
pr_validate <- map_dfr(increments, ~make_pr_curve(., for_pr_validate))  %>%
  mutate(set = "validate")

pr <- bind_rows(pr_train, pr_validate)


pr_plot <- pr %>%
   ggplot() + 
  geom_line(aes(x = recall, y = precision, color = set), alpha = 0.7, size = 2) + 
  scale_color_manual(values = c("blue", "red")) +
  xlim(0,1) +
  ylim(0,1) +
  geom_label(x = 0.85, y = 0.9, label = "train", color = "blue", size = 3) +
  geom_label(x = 0.5, y = 0.7, label = "validate", color = "red", size = 3)   +
  theme(legend.position = "none")
pr_plot
  

pr_plot %>% ggsave("figures/pr_plot.png", .,device = "png", width = 2.5, height = 2.5,  units = "in")
pr_plot %>% ggsave("figures/pr_plot.pdf", .,device = "pdf", width = 2.5, height = 2.5,  units = "in")

```

FDR plot
```{r}
fdr_plot <- pr %>%
   ggplot() + 
  geom_line(aes(x = cutoff, y = precision, color = set), alpha = 0.7, size = 2) + 
  scale_color_manual(values = c("blue", "red")) +
  ylim(0,1) +
  ylab("1-FDR") +
  geom_hline(yintercept = 0.9) +
  geom_hline(yintercept = 0.5) +  
  geom_label(x = 0.85, y = 0.9, label = "train", color = "blue", size = 3) +
  geom_label(x = 0.5, y = 0.7, label = "validate", color = "red", size = 3)   +
  theme(legend.position = "none")
fdr_plot
  

fdr_plot %>% ggsave("figures/fdr_plot.png", .,device = "png", width = 2.5, height = 2.5,  units = "in")
fdr_plot %>% ggsave("figures/fdr_plot.pdf", .,device = "pdf", width = 2.5, height = 2.5,  units = "in")

```



Are fragments enriched in particular domains?
```{r}




# Graph formatting for sugiyama plot
fragment_domains_setup <- fragment_domains_auto %>%  #fragment_domains %>% 
  filter(grepl("HUMAN", ProteinID)) %>%
  select(Entry_name, Annotation ) %>%
  filter(!is.na(Annotation)) %>%
    filter(Annotation!= "nodomain") %>%
  unique() %>%
  as_tbl_graph(from = Entry_name, to = Annotation) %>%
  activate("nodes") %>%
  mutate(nodecolor_all = case_when(
       grepl("_HUMAN", name) ~ "HUMAN",
       grepl("_CHLRE", name) ~ "CHLRE",
       grepl("_ARATH", name) ~ "ARATH",
       TRUE ~ "Domain"
  )) %>%
  mutate(labelleft = case_when(
       grepl("_[HUMAN|ARATH|CHLRE]", name) ~ str_replace(name, "_HUMAN", "")
  )) %>%
    mutate(labelright = case_when(
       !grepl("_[HUMAN|ARATH|CHLRE]", name) ~ name
  )) 

fragment_domains_human_plot <- fragment_domains_setup %>%
  
  ggraph(layout = "sugiyama") +
    geom_edge_link() +
    geom_node_point(aes(color = nodecolor_all))+
  geom_node_text(aes(label = labelright), hjust = 0, nudge_y = 0.1) +
  geom_node_text(aes(label = labelleft), hjust = 1, nudge_y = -0.1) +
 # coord_flip() +
  scale_color_manual(values = c( "orange", "blue")) +
  scale_y_reverse(limits = c(3, -3)) +
  theme(axis.text.x = element_blank(),
        axis.title.x = element_blank(),
        axis.text.y = element_blank(),
        axis.title.y = element_blank(),
        legend.position = "none")
  
fragment_domains_human_plot %>% ggsave("figures/fragment_domains_human_plot.png", .,device = "png", width = 7, height = 20,  units = "in")
fragment_domains_human_plot %>% ggsave("figures/fragment_domains_human_plot.pdf", ., device = "pdf", width = 7, height = 20, units = "in")




fragment_domains_setup_multiprot <- 
  fragment_domains_auto %>%
  select(Entry_name, Annotation ) %>%
  unique() %>%
   filter(!is.na(Annotation)) %>%
    filter(Annotation!= "nodomain") %>%
  group_by(Annotation) %>%
    mutate(n = n())%>% ungroup %>%
  filter(n > 1) %>%
  as_tbl_graph(from = Entry_name, to = Annotation) %>%
  activate("nodes") %>%
  mutate(nodecolor_all = case_when(
       grepl("_HUMAN", name) ~ "HUMAN",
       grepl("_CHLRE", name) ~ "CHLRE",
       grepl("_ARATH", name) ~ "ARATH",
       TRUE ~ "Domain"
  )) %>%
  mutate(labelleft = case_when(
       grepl("_[HUMAN|ARATH|CHLRE]", name) ~ name
  )) %>%
    mutate(labelright = case_when(
       !grepl("_[HUMAN|ARATH|CHLRE]", name) ~ name
  )) 

fragment_domains_multispec_plot <- fragment_domains_setup_multiprot %>%
  
  ggraph(layout = "sugiyama") +
    geom_edge_link() +
    geom_node_point(aes(color = nodecolor_all))+
  geom_node_text(aes(label = labelright), hjust = 0, nudge_y = 0.1) +
  geom_node_text(aes(label = labelleft, color = nodecolor_all), hjust = 1, nudge_y = -0.1) +
  coord_flip() +
  scale_color_manual(values = c("orange", "darkgreen",  "black", "blue")) +
  scale_y_reverse(limits = c(4, -3)) +
  theme(axis.text.x = element_blank(),
        axis.title.x = element_blank(),
        axis.text.y = element_blank(),
        axis.title.y = element_blank(),
        legend.position = "none")
  
fragment_domains_multispec_plot %>% ggsave("figures/fragment_domains_multispec_plot.png", .,device = "png", width = 8, height = 20,  units = "in")
fragment_domains_multispec_plot %>% ggsave("figures/fragment_domains_multispec_plot.pdf", ., device = "pdf", width = 8, height = 20, units = "in")

```



Get N-fragments that end before first domain
Get C-fragments that start after last domain 

```{r}
proteins_w_full <- fragment_data %>%
  filter(term== "full") %>% select(ProteinID) %>% unique

# Not good, see INT3_HUMAN
experiments_w_full <- fragment_data %>%
  filter(term== "full") %>% select(ProteinID, experiment_name) %>% unique


fragment_data_terminal %>% filter(grepl("INT3_", ProteinID))

```


```{r}
bps_pfam <- fragment_data_terminal %>%
   filter(!is.na(loc)) %>%

  inner_join(domains_pfam, by = "Entry") %>%
      filter(!is.na(Source_ID)) %>%
  mutate(in_domain = case_when((loc >= ProtBegin & loc <= ProtEnd) ~ TRUE,
                                TRUE ~ FALSE)) %>% 
  arrange(desc(seqlen))


  
domain_plot <- bps_pfam %>%
  filter(grepl("HUMAN", ProteinID)) %>% 
  #filter(!is.na(Entry_name)) %>% 
 # mutate(Entry_name = fct_inorder(Entry_name)) %>%
  #filter(Entry_name == "DHB4_HUMAN") %>%
  filter(term %in% c("nterm", "cterm")) %>%
  ggplot() + 
    geom_rect(fill = "grey90", aes(xmin = 0, xmax = seqlen, ymin = 0, ymax = 1)) +
  geom_rect(fill = "#97b6b9", aes(xmin = ProtBegin, xmax = ProtEnd, ymin = 0, ymax = 1)) +
  geom_point(color = "red", aes(x = loc, y =0.5, shape = term), size = 2.5) +
  #geom_point(color = "red", aes(x = loc, y =0.5, shape = direction)) +
  scale_shape_manual(values = c(91,93, NA)) +
  #geom_point(shape=18, color = "red", aes(x = loc, y =0.5)) + 
  #geom_vline(color  = "black", xintercept = 0) +
   # geom_vline(color = "black", aes(xintercept = seqlen)) + 
  theme(panel.margin=unit(.05, "lines"),
        strip.text.y.left = element_text(angle = 0),
        #strip.text.y = element_text(angle = 180),
        strip.background =element_rect(fill = "white"),
       axis.text.x = element_blank(), 
        axis.ticks.x = element_blank(),
       axis.text.y = element_blank(),
       axis.ticks.y = element_blank(),
       axis.line.x = element_blank(),
       axis.line.y = element_blank(),
       axis.title.x = element_blank(),
       axis.title.y = element_blank()
       #panel.border = element_rect(color = "black", fill = NA, size = 1)
        ) +
          facet_wrap(~fct_inorder(Entry_name), strip.position = "left",ncol = 1) 
domain_plot


domain_plot %>% ggsave("figures/domain_plot.png", .,device = "png", width = 7, height = 10,  units = "in")
domain_plot %>% ggsave("figures/domain_plot.pdf", ., device = "pdf", width = 7, height = 10, units = "in")
```

```{r}
domains_pfam %>%

      filter(!is.na(Source_ID)) %>%
  filter(Entry %in% fragment_data_terminal$Entry) %>%
  rowwise %>%
  mutate(domlength = ProtEnd - ProtBegin) %>%
  select(Entry, domlength) %>%
  group_by(Entry) %>%
     summarize(tot_dom_cov = sum(domlength)) %>%
  ungroup() %>%
  left_join(meta ) %>%
      filter(grepl("HUMAN", Entry_name)) %>%
   mutate(prop_domain = tot_dom_cov/seqlen) %>%
   summarize(x = 1- mean(prop_domain))

bps_pfam %>%

  filter(grepl("HUMAN", ProteinID)) %>% 
  select(ProteinID, peakid, term, in_domain) %>%
  filter(term %in% c("nterm", "cterm")) %>%
  unique() %>%
  group_by(in_domain) %>%
     summarize(n = n())


```

????
```{r}

domain_coverage <- bps_pfam %>%

  rowwise() %>%
  mutate(dom_length = ProtEnd-ProtBegin) %>%
  group_by(Entry_name, seqlen) %>%
    summarize(tot_dom_length = sum(dom_length)) %>%
  ungroup %>%
  rowwise() %>%
  mutate(dom_coverage = tot_dom_length/seqlen) %>%
  mutate(nondomain = 1-dom_coverage)


prod(domain_coverage$nondomain, na.rm = TRUE)

```






```{r}
bps_dis <- fragment_data_terminal %>%
  filter(!is.na(loc)) %>%
  left_join(meta) %>% #, by = "Entry") %>%
  inner_join(domains_dis, by = "Entry") %>%
  mutate(in_dis = case_when((loc >= ProtBegin & loc <= ProtEnd) ~ TRUE,
                                TRUE ~ FALSE)) %>% 
  arrange(seqlen)
  
dis_plot <- bps_dis %>%
  filter(experiment_name == "CHLRE_IEX1") %>%
  mutate(Entry_name = fct_inorder(Entry_name)) %>%
  #filter(Entry_name == "UBP15_HUMAN") %>%
  ggplot() + 
    geom_rect(fill = "grey90", aes(xmin = 0, xmax = seqlen, ymin = 0, ymax = 1)) +
  geom_rect(fill = "cadetblue4", aes(xmin = ProtBegin, xmax = ProtEnd, ymin = 0, ymax = 1)) +
  geom_point(color = "red", aes(x = loc, y =0.5, shape = term)) +
   scale_shape_manual(values = c(91,93)) +
  #geom_point(shape=18, color = "red", aes(x = loc, y =0.5)) + 
  #geom_vline(color  = "black", xintercept = 0) +
   # geom_vline(color = "black", aes(xintercept = seqlen)) + 
  theme(panel.margin=unit(.05, "lines"),
        strip.text.y.left = element_text(angle = 0),
        strip.background =element_rect(fill = "white"),
       axis.text.x = element_blank(), 
        axis.ticks.x = element_blank(),
       axis.text.y = element_blank(),
       axis.ticks.y = element_blank(),
       axis.line.x = element_blank(),
       axis.line.y = element_blank(),
       axis.title.x = element_blank(),
       panel.border = element_rect(color = "black", fill = NA, size = 1)
        ) +
           facet_wrap(~Entry_name, strip.position = "left",ncol = 2) 
dis_plot


#domain_plot %>% ggsave("figures/domain_plot.png", .,device = "png", width = 7, height = 5,  units = "in")
#domain_plot %>% ggsave("figures/domain_plot.pdf", ., device = "pdf", width = 7, height = 5, units = "in")


#Look up BTF3 and NACA
# Very important - Why no picture of NACA generated?

```





Are fragments more disordered than rest of prot?
```{r}

diso_coverage <- domains_dis  %>%
  mutate(length = ProtEnd - ProtBegin) %>%
  group_by(Entry) %>%
     summarize(totaldis = sum(length)) %>%
  ungroup %>%
  inner_join(meta) %>%
  filter(!is.na(seqlen))

# bps_dis <- fragment_data_terminal %>% ungroup %>%
#  inner_join(meta) %>%
#  filter(!is.na(seqlen)) %>%
#  mutate(fragment_start = 
#           case_when(term == "nterm" ~ 1,
#                     term == "cterm" ~ loc)) %>%
#  mutate(fragment_end = 
#           case_when(term == "nterm" ~ as.integer(loc),
#                     term == "cterm" ~ seqlen)) %>%  
#  left_join(domains_dis)
 
 
bps_dis_coverage <- bps_dis %>%
  select(ProteinID, Entry, nterm_pos,cterm_pos, ProtBegin, ProtEnd) %>%
  mutate(dis_frag_start = case_when(
    ProtBegin >= nterm_pos  & ProtBegin <= cterm_pos ~ ProtBegin,
    cterm_pos >= ProtBegin & nterm_pos <= ProtEnd ~  as.integer(nterm_pos)
    
  ))   %>%
  filter(!is.na(dis_frag_start)) %>%
  mutate(dis_frag_end = case_when(
    cterm_pos >= ProtEnd ~ ProtEnd,
    cterm_pos <= ProtEnd ~ cterm_pos
       
  )) %>%
  mutate(fragment_tot_length = cterm_pos - nterm_pos) %>%
  mutate(dis_length = dis_frag_end - dis_frag_start) %>%
    left_join(diso_coverage) %>%
  mutate(fragment_dis_cov = dis_length/fragment_tot_length,
         total_dis_cov = totaldis/seqlen) %>%
  mutate(cov_diff = fragment_dis_cov/ total_dis_cov) 
  
bps_dis_coverage %>%
  ggplot(aes(x = total_dis_cov, y = fragment_dis_cov, size = cov_diff, label = ProteinID)) + 
  geom_point()
  
```

Get length of each disordered terminal. Each protein has up to two disordered terminal
Compare to protein abundance as well
```{r}


per_prot_abundance <- long_sel %>%
  filter(experiment_name != "HEKB_SEC1") %>%
  group_by(ProteinID) %>%
     summarize(tot = sum(pepcount)) %>% ungroup %>%
  separate(ProteinID, into = c(NA, "Entry", NA), sep = "[|]")
  
all_disordered_termini <- domains_dis %>% left_join(meta, by = "Entry") %>%
  filter(ProtBegin ==1 |  ProtEnd == seqlen) %>%
  mutate(dis_length = ProtEnd - ProtBegin) %>%
  select(Entry, dis_length) %>%
  mutate(set = "all disordered termini") 

fragment_disordered_termini <-bps_dis_coverage %>%
  filter(dis_frag_start ==1 | dis_frag_end == seqlen) %>%
  select(Entry, dis_length) %>%
  mutate(set = "fragment disordered_termini")

both_disordered_termini <- bind_rows(
  all_disordered_termini, 
  fragment_disordered_termini
) %>%
 
  left_join(per_prot_abundance, by = "Entry")


both_disordered_termini %>%
  
  ggplot(aes(x = tot, fill = set)) +
    geom_histogram(position = "dodge") + 
  #facet_wrap(~set,  ncol = 1) +
  scale_x_log10() +
  scale_y_log10(expand = c(0,0))
  
  
#    ggplot(aes(x = tot, y = fragment_dis_length, size = fragment_dis_cov)) +
#  geom_point() +
#  scale_x_log10()

bps_loc %>% ungroup %>%
  select(ProteinID, Entry) %>%
  left_join(meta) %>% View()
```




UNUSED
```{r}
theme_set(theme_cowplot_consistent_text())

fisher_output_summary %>% 
  inner_join(scored_labeled_all, by = c("ProteinID", "experiment_name")) %>%
  inner_join(labels) %>%
  left_join(too_few_peptides) %>%
  mutate(p_value_bh = case_when(toofew == TRUE ~ 100,
                                TRUE ~ p_value_bh)) %>%
   ggplot(aes(x = max_abs_log2fc, y = p_value_bh)) + 
      geom_point(alpha = 0.05) + 
  scale_y_log10() + 
  geom_vline(xintercept = 3.5, linetype = "dashed") + 
  geom_hline(yintercept = 0.01, linetype = "dashed") +
  facet_wrap(~label, ncol = 1)



scored_labeled_all %>%
  left_join(labels) %>%
   inner_join(fisher_output_summary, by = c("ProteinID", "experiment_name")) %>%
  ggplot(aes(x = max_abs_log2fc, y = min_p_value)) + 
      geom_point(alpha = 0.1) + 
  scale_y_log10() + 
  facet_wrap(~label, ncol = 1)


theme_set(theme_cowplot_consistent_text(font_size = 8))
# With abundance matching
a <- scored_labeled_all_formatted %>%
  #mutate(custom_label = fct_relevel(custom_label, c("Positive train", "Positive validate", "Negative train","Negative validate" ))) %>%
  ggplot(aes(x = max_abs_log2fc, fill = custom_label)) + 
  geom_histogram(binwidth = 0.1) + 
  background_grid(major = ("x")) +
  #scale_x_log10()+ 
  #scale_fill_manual(values= c(palette[2], "grey")) +
  #xlab("Proteoform score") + 
  xlab("Max absolute log2 Fold Change\n(N vs. C terminal observed/observable peptides") +
 # xlab("Max abs") +
  ylab("# of proteins") +
  theme(legend.position = "none", strip.background = element_blank()) +
  facet_wrap(~custom_label,scales= "free_y", ncol = 1)


fragment_data_final <- fragment_data %>% filter(term != "full") %>%
  select(ProteinID, experiment_name, term) %>% unique()


theme_set(theme_cowplot_consistent_text(font_size = 8))
# With abundance matching
b <- fisher_output_summary_formatted %>%
  #mutate(custom_label = fct_relevel(custom_label, c("Positive train", "Positive validate", "Negative train","Negative validate" ))) %>%
  ggplot(aes(x = -log10(min_p_value), fill = custom_label)) + 
  geom_histogram(binwidth = 1) + 
  background_grid(major = ("x")) +
  #scale_x_log10()+ 
  #scale_fill_manual(values= c(palette[2], "grey")) +
  #xlab("Proteoform score") + 
 # xlab("Max absolute log2 Fold Change\n(N vs. C terminal observed/observable peptides") +
  xlab("-log10(p.value)") +
  ylab("# of proteins") +
  theme(legend.position = "none", strip.background = element_blank()) +
  facet_wrap(~custom_label,scales= "free_y", ncol = 1)


plot_grid(a,b)


fisher_output_summary_formatted %>% inner_join(scored_labeled_all_formatted) %>%
  ggplot(aes(x = -log10(min_p.value), y = max_abs_log2fc, fill = custom_label)) + 
  geom_point(alpha = 0.1) +
  #geom_histogram(binwidth = 1) + 
  #background_grid(major = ("x")) +
  #scale_x_log10()+ 
  #scale_fill_manual(values= c(palette[2], "grey")) +
  #xlab("Proteoform score") + 
 # xlab("Max absolute log2 Fold Change\n(N vs. C terminal observed/observable peptides") +
  xlab("-log10(p.value)") +
  ylab("# of proteins") +
  theme(legend.position = "none", strip.background = element_blank()) +
  facet_wrap(~custom_label,scales= "free_y", ncol = 1) +
  geom_hline(yintercept = 3.5, linetype = "dashed") +
  geom_vline(xintercept = 8, linetype = "dashed")

theme_set(theme_cowplot_consistent_text(font_size = 8))

fisher_output_summary%>%
  left_join(labels) %>%
  
  
  #mutate(custom_label = fct_relevel(custom_label, c("Positive train", "Positive validate", "Negative train","Negative validate" ))) %>%
  ggplot(aes(x = -log10(min_p.value), fill = custom_label)) + 
  geom_histogram(binwidth = 0.1) + 
  background_grid(major = ("x")) +
  scale_x_log10()+ 
  scale_fill_manual(values= c(palette[2], "grey")) +
  #xlab("Proteoform score") + 
 # xlab("Max absolute log2 Fold Change\n(N vs. C terminal observed/observable peptides") +
  xlab("Minimum p-value") +
  ylab("# of proteins") +
  theme(legend.position = "none", strip.background = element_blank()) +
  facet_wrap(~custom_label,scales= "free_y", ncol = 1)




```


```{r}


scored_labeled_all_formatted <- scored_labeled_all %>%
  left_join(labels) %>%
  #left_join(too_few_peptides) %>%
  #mutate(p_value_bh = case_when(toofew == TRUE ~ 1000, # Set very high p-value for too low abundance to be fair in match to the max_abs_log2fc case
                #                TRUE ~ p_value_bh)) %>%
  
  #mutate(custom_label = case_when(label == "neg" & set == "train" ~ "Negative train",
  #                         label == "pos" & set == "train" ~ "Positive train",
  #                         label == "neg" & set == "val" ~ "Negative validate",
  #                         label == "pos" & set == "val" ~ "Positive validate"
  #                         
  #                         )) %>%
    left_join(fragment_data_final) %>%
  
 
  mutate(custom_label = case_when(label=="pos" ~ "Hand selected truncated proteins",
                                  label == "neg" ~ "Random proteins",
                                  !is.na(term) ~ "Final selection",
                                  TRUE ~ "MORE")) %>%
   filter(!is.na(custom_label)) 



fisher_output_summary_formatted <- fisher_output_summary %>%
  left_join(labels) %>%
  left_join(fragment_data_final) %>%
  left_join(too_few_peptides) %>%
  mutate(p_value_bh = case_when(toofew == TRUE ~ 1000, # Set very high p-value for too low abundance to be fair in match to the max_abs_log2fc case
                                TRUE ~ p_value_bh)) %>%
  

  
 
  mutate(custom_label = case_when(label=="pos" ~ "Hand selected truncated proteins",
                                  label == "neg" ~ "Random proteins",
                                  !is.na(term) ~ "Final selection",
                                  is.na(label) ~ "MORE")) %>%
   filter(!is.na(custom_label)) 



allscored_formatted <- fisher_output_summary_formatted %>% full_join(scored_labeled_all_formatted)  %>%
  mutate(max_abs_log2fc= case_when(is.na(max_abs_log2fc) ~ 0, TRUE ~ max_abs_log2fc)) %>%
  select(-peak, -midpoint, -experiment_name, -data, -estimate, -p.value, -conf.low, -conf.high, -method, -alternative, -term, -actual_terminal_peptide, -terminal, -toofew) %>%
  unique() 

theme_set(theme_cowplot_consistent_text(font_size = 8))

a <- allscored_formatted %>%
    filter(custom_label %in% c("Hand selected truncated proteins", "Random proteins")) %>%
    ggplot(aes(x = -log10(min_p_value), fill = custom_label)) + 
  geom_histogram(binwidth = 0.4) + 
  background_grid(major = ("x")) +
  #geom_density() + 
  #scale_x_log10()+ 
  scale_fill_manual(values= c(palette[2], "grey")) +
  #xlab("Proteoform score") + 
  xlab("Min -log10(p-value)") +
 # xlab("Max absolute log2 Fold Change\n(N vs. C terminal observed/observable peptides") +
  #xlab("Minimum p-value") +
  ylab("# of proteins") +
  theme(legend.position = "none", strip.background = element_blank()) +
  facet_wrap(~custom_label,scales= "free_y", ncol = 1)


b <- allscored_formatted%>%
  filter(custom_label %in% c("Hand selected truncated proteins", "Random proteins")) %>%
    ggplot(aes(x = max_abs_log2fc, fill = custom_label)) + 
  geom_histogram(binwidth = 0.2) + 
  #geom_density() + 
  background_grid(major = ("x")) +
  #scale_x_log10()+ 
  scale_fill_manual(values= c(palette[2], "grey")) +
  #xlab("Proteoform score") + 
  xlab("Max terminal bias") + 
 # xlab("Max absolute log2 Fold Change\n(N vs. C terminal observed/observable peptides") +
  #xlab("Minimum p-value") +
  ylab("# of proteins") +
  theme(legend.position = "none", strip.background = element_blank()) +
  facet_wrap(~custom_label,scales= "free_y", ncol = 1)


# Actually with assemble with cowplot
new_Figure2b<- plot_grid(b,a, ncol = 1)
new_Figure2b
new_Figure2b %>% ggsave("figures/new_Figure2b.png", .,device = "png", width = 2.5, height = 3.5,  units = "in")
new_Figure2b %>% ggsave("figures/new_Figure2b.pdf", .,device = "pdf", width = 2.5, height = 3.5,  units = "in")

```

UNUSED
```{r}



c <- allscored_formatted%>%
  filter(custom_label %in% c("Hand selected truncated proteins", "Random proteins")) %>%
    ggplot(aes(x = max_abs_log2fc * -log10(min_p_value), fill = custom_label)) + 
  #geom_histogram(binwidth = 1) + 
  geom_density() +
  background_grid(major = ("x")) +
  #scale_x_log10()+ 
  scale_fill_manual(values= c(palette[2], "grey")) +
  #xlab("Proteoform score") + 
  xlab("Max score") +
 # xlab("Max absolute log2 Fold Change\n(N vs. C terminal observed/observable peptides") +
  #xlab("Minimum p-value") +
  ylab("# of proteins") +
  theme(legend.position = "none", strip.background = element_blank()) +
  facet_wrap(~custom_label,scales= "free_y", ncol = 1) 

```

```{r}
palette <- rep(c("#0072B2","#E69F00","#009E24","#FF0000", "#979797","#5530AA"), 8)

theme_set(theme_cowplot_consistent_text(font_size = 8))

# Supplementary Figure 1
new_SuppFigure1 <- allscored_formatted%>%
    filter(custom_label %in% c("Hand selected truncated proteins", "Random proteins")) %>%
   mutate(g = case_when(min_p_value < 0.001 & max_abs_log2fc > 3  ~ "Both",
                            min_p_value < 0.001 ~ "P-value threshold",
                            max_abs_log2fc > 3 ~ "Terminal bias threshold", 
                            TRUE ~ "Neither")) %>%
  
  mutate(g = factor(g, levels = c("P-value threshold", "Neither", "Both", "Terminal bias threshold"))) %>%


   ggplot(aes(x = max_abs_log2fc, y =-log10(min_p_value), color = g)) + #, color = fct_rev(fct_relevel(g, "Neither",  after = Inf)))) +
   xlab("Max terminal bias") +
   ylab("Min -log10(p-value)") +
   geom_point(alpha = 0.5)+
  theme( strip.background = element_blank()) +
  theme(legend.position = "bottom", legend.title = element_blank()) +
 
  facet_wrap(~custom_label, ncol = 1) +
  guides(color = guide_legend(nrow = 2, title.position = "top")) +
  geom_hline(yintercept = -log10(0.001), linetype = "dashed", color = palette[1]) +
  geom_vline(xintercept = 3, linetype = "dashed", color = palette[2]) +
  scale_color_manual(values = c("P-value threshold" = "#0072B2", 
                                "Both" = "#009E24", 
                                "Neither" = "#979797", 
                                "Terminal bias threshold" = "#E69F00"),
                     name = "Prioritized")



new_SuppFigure1 
new_SuppFigure1 %>% ggsave("figures/new_SuppFigure1.png", .,device = "png", width = 3.5, height = 4.5,  units = "in")
new_SuppFigure1 %>% ggsave("figures/new_SuppFigure1.pdf", .,device = "pdf", width = 3.5, height = 4.5,  units = "in")

```

UNUSED
```{r}

# Replace Figure 2B
 allscored_formatted%>%
  filter(custom_label %in% c("Hand selected truncated proteins", "Random proteins")) %>%
    ggplot(aes(x = max_abs_log2fc * -log10(min_p_value), fill = custom_label)) + 
  geom_histogram(binwidth = 1) + 
  #geom_density() +
  background_grid(major = ("x")) +
  #scale_x_log10()+ 
  scale_fill_manual(values= c(palette[2], "grey")) +
  #xlab("Proteoform score") + 
  xlab("Max score") +
 # xlab("Max absolute log2 Fold Change\n(N vs. C terminal observed/observable peptides") +
  #xlab("Minimum p-value") +
  ylab("# of proteins") +
  theme(legend.position = "none", strip.background = element_blank()) +
  facet_wrap(~custom_label,scales= "free_y", ncol = 1) 




p <- allscored_formatted%>%
    filter(custom_label %in% c("Hand selected truncated proteins", "Random proteins")) %>%
   mutate(g = case_when(min_p_value < 0.001 & max_abs_log2fc > 3  ~ 0,
                            min_p_value < 0.001 ~ 1,
                             max_abs_log2fc > 3 ~ 2, 
                            TRUE ~ 3)) %>%
   ggplot(aes(x = max_abs_log2fc, y =-log10(min_p_value), color = label, group = label)) + 
   geom_point(alpha = 0.2)#+
  # scale_color_manual(values= c(palette[2], palette[1], palette[3], palette[5])) 

library(ggExtra)
ggMarginal(p, type = "density", size = 5,groupColour = TRUE, groupFill = TRUE)

```

```{r}
library(ggridges)
allscored_formatted%>%
    filter(custom_label %in% c("Hand selected truncated proteins", "Random proteins")) %>%
    ggplot(aes(x = -log10(min_p_value), group = custom_label)) + #, y = max_abs_log2fc, group = custom_label)) + #fill = custom_label)) + 
 geom_density_ridges() +
  #background_grid(major = ("x")) +
  #scale_x_log10()+ 
  #scale_fill_manual(values= c(palette[2], "grey")) +
  #xlab("Proteoform score") + 
 # xlab("Max absolute log2 Fold Change\n(N vs. C terminal observed/observable peptides") +
  xlab("Minimum p-value") +
  ylab("# of proteins") +
  theme(legend.position = "none", strip.background = element_blank()) #+
  #facet_wrap(~custom_label,scales= "free_y", ncol = 1)


b <- allscored_formatted%>%
  filter(custom_label %in% c("Hand selected truncated proteins", "Random proteins")) %>%
    ggplot(aes(x = max_abs_log2fc, fill = custom_label)) + 
  geom_histogram(binwidth = 0.2) + 
  background_grid(major = ("x")) +
  #scale_x_log10()+ 
  scale_fill_manual(values= c(palette[2], "grey")) +
  #xlab("Proteoform score") + 
 # xlab("Max absolute log2 Fold Change\n(N vs. C terminal observed/observable peptides") +
  #xlab("Minimum p-value") +
  ylab("# of proteins") +
  theme(legend.position = "none", strip.background = element_blank()) +
  facet_wrap(~custom_label,scales= "free_y", ncol = 1)


```

```{r}
score_dist_plot
  

fisher_output_summary_formatted %>% full_join(scored_labeled_all_formatted)  %>%
  mutate(max_abs_log2f= case_when(is.na(max_abs_log2fc) ~ -1, TRUE ~ max_abs_log2fc)) %>%
  ggplot(aes(x = -log10(min_p_value), y = max_abs_log2fc)) +
  geom_point(alpha =0.1)+
  facet_wrap(~custom_label)

```


```{r}



# This is too slow to run in Rstudio, migrating to script
library(furrr)
library(future)
plan(strategy = multisession)
contingency_full_nested   <- all_combinations_fixed %>%
  ungroup() %>% 
  group_by(ProteinID, experiment_name, peak, midpoint) %>%
  nest()
 

contingency_full_nested_sel <- contingency_full_nested %>% filter(ProteinID %in% fragment_data_terminal$ProteinID)



 contingency_full_nested_sel%>% 
   filter(grepl("SHOT1", ProteinID)) %>%
   select(peak, midpoint)
 
 
 ```
 
 
 ```{r}
 
 fisher_output_sel <- contingency_full_nested_sel %>%  
     mutate(test_result =  map(data, ~broom::tidy(fisher.test(xtabs(n ~ pep_half + presence, data = .))), .progress = TRUE)) %>%
  unnest(test_result)
 

fisher_output_sel %>% write_csv("fisher_output_sel.csv")
 
 
fisher_output_full <- contingency_full_nested %>%  
     mutate(test_result =  map(data, ~broom::tidy(fisher.test(xtabs(n ~ pep_half + presence, data = .))), .progress = TRUE)) %>%
  unnest(test_result)


fisher_output_full %>% write_csv("fisher_output_full.csv")

```

unused
```{r}


contingency <- short_tidy_unique_peaks_expanded_fragments %>% #filter(grepl("UBP15", ProteinID)) %>%
  separate(ProteinID, into = c("tmp", "Entry", "Entry_name"), sep= "[|]",remove = FALSE) %>%
  select(-seqlen) %>%
  left_join(meta) %>%
  mutate(midpoint = seqlen/2) %>%
  mutate(pep_half = case_when(End <= midpoint ~ "firsthalf", 
                              TRUE ~ "secondhalf"))  %>%
  select(ProteinID, Peptide, experiment_name, peak, pep_half, presence) %>% #filter(grepl("UBP15", ProteinID)) %>%
  group_by(ProteinID, experiment_name, peak, pep_half, presence) %>%
    summarize(n = n()) %>%
    arrange(peak, pep_half, presence)  %>%
    group_by(ProteinID, experiment_name, peak) %>%
  complete(pep_half = c("firsthalf", "secondhalf"), 
           presence = c("missing", "observed"), 
           fill = list(n = 0))



fisher_output <- contingency %>%
  ungroup() %>% 
  group_by(ProteinID, experiment_name, peak) %>%
  nest() %>%
    mutate(test_result = map(data, ~broom::tidy(fisher.test(xtabs(n ~ pep_half + presence, data = .))))) %>%
  unnest(test_result)


fisher_output %>% arrange(p.value) %>% head()

tail(fisher_output)



```

```{r}
fisher_output

for_pr_fisher <- fisher_output %>%
  
  left_join(labels) %>%
  mutate(id = paste0(ProteinID, "-", experiment_name)) %>%
  mutate(score= -log(p.value)) %>%
  select(id, label, set, score) %>%
  mutate(num_label = case_when(
    label == "pos" ~ 1,
    label == "neg" ~ -1
  )) 

for_pr_train_fisher <- for_pr_fisher %>%
  filter(set == "train")
for_pr_validate_fisher <- for_pr_fisher %>%
  filter(set == "val")

increments_fisher <- seq(0, max(for_pr_fisher$score), 0.1)
pr_train_fisher <- map_dfr(increments_fisher, ~make_pr_curve(., for_pr_train_fisher)) %>%
  mutate(set = "train")
pr_validate_fisher <- map_dfr(increments_fisher, ~make_pr_curve(., for_pr_validate_fisher))  %>%
  mutate(set = "validate")

pr_fisher <- bind_rows(pr_train_fisher, pr_validate_fisher)


pr_plot_fisher <- pr_fisher %>%
   ggplot() + 
  geom_line(aes(x = recall, y = precision, color = set), alpha = 0.7, size = 2) + 
  scale_color_manual(values = c("blue", "red")) +
  xlim(0,1) +
  ylim(0,1) +
  geom_label(x = 0.85, y = 0.9, label = "train", color = "blue", size = 3) +
  geom_label(x = 0.5, y = 0.7, label = "validate", color = "red", size = 3)   +
  theme(legend.position = "none")
pr_plot_fisher
  

pr_plot %>% ggsave("figures/pr_plot.png", .,device = "png", width = 2.5, height = 2.5,  units = "in")
pr_plot %>% ggsave("figures/pr_plot.pdf", .,device = "pdf", width = 2.5, height = 2.5,  units = "in")



```

# Unused
```{r}
exps_w_fl <- fragment_data%>%
   inner_join(over2_peptide_counts) %>%
  filter(!experiment_name %in% c("HEKR_IEX1", "HEKR_IEX2", "HEKB_SEC1")) %>%
  filter(term == "full") %>%
  select(ProteinID, experiment_name) %>%
  unique()

exps_w_terms <- fragment_data%>%
  filter(term %in% c("cterm", "nterm")) %>%
  select(ProteinID, experiment_name) %>% unique()

exps_w_both <- inner_join(exps_w_fl, exps_w_terms) %>%
  filter(grepl("SEC", experiment_name))

locs_fl_vs_frag <- fragment_data %>% semi_join(exps_w_both) %>% #  head %>% 
  group_by(ProteinID, experiment_name, peakid, term) %>%
      summarize(mid_frac = weighted.mean(FractionOrder, pepcount))
locs_fl<- locs_fl_vs_frag %>% filter(term == "full")
locs_frag <- locs_fl_vs_frag %>% filter(term != "full")

locs_both <- locs_fl %>% full_join(locs_frag, by = c("ProteinID", "experiment_name"), multiple = "all", suffix = c("full", "frag"))

locs_both %>% filter(mid_fracfrag == mid_fracfull)

locs_both_annot <- locs_both %>%
  filter(grepl("HEK", experiment_name)) %>%
  mutate(sizebal = case_when(mid_fracfull < mid_fracfrag ~ "expected",
                             mid_fracfull > mid_fracfrag ~ "unexpected")) 

locs_both_annot %>%
  filter(sizebal == "unexpected")
locs_both_annot %>%
  group_by(sizebal) %>%
     summarize(n= n())

locs_both %>%  ggplot(aes(x = mid_fracfull,  y = mid_fracfrag, label = ProteinID, color = termfrag)) + 
  geom_point(alpha= 0.2) +
  geom_abline(intercept = 0, slope = 1) 
```



```{r}
breakpoints_test <- breakpoints %>% filter(grepl("SHOT1", ProteinID) | grepl("S23IP", ProteinID))
short_tidy_unique_peaks_expanded_fragments_test <- short_tidy_unique_peaks_expanded_fragments  %>% filter(grepl("SHOT1", ProteinID) | grepl("S23IP", ProteinID))

contingency_full_step1_test <- short_tidy_unique_peaks_expanded_fragments_test %>% 
  left_join(breakpoints_test, by= "ProteinID") %>%
  mutate(pep_half = if_else(End <= midpoint, "firsthalf", "secondhalf")) %>%

  #mutate(pep_half = case_when(End <= midpoint ~ "firsthalf", 
  #                            TRUE ~ "secondhalf"))  %>%
  select(ProteinID, Peptide, experiment_name, peak, pep_half, presence, midpoint) %>%
  group_by(ProteinID, experiment_name, peak, pep_half, presence, midpoint) %>%
    summarize(n = n())



all_combinations_test <- with(contingency_full_step1_test, 
                         expand.grid(ProteinID = unique(ProteinID),
                                     experiment_name = unique(experiment_name),
                                     peak = unique(peak),   
                                     midpoint = unique(midpoint),
                                     pep_half = c("firsthalf", "secondhalf"),
                                     presence = c("missing", "observed")))


contingency_full_step1_test %>% 
  group_by(ProteinID, peak, midpoint, experiment_name) %>%
  complete(pep_half = c("firsthalf", "secondhalf"), 
                  presence = c("missing", "observed"), 
                  fill = list(n = 0)) 
 
all_combos_test2 <- contingency_full_step1_test %>%
  group_by(ProteinID, peak, midpoint, experiment_name) %>%
    complete( pep_half = c("firsthalf", "secondhalf"), presence = c("missing", "observed"), fill = list(n = 0))

all_combos_test2

all_combos_test2 %>%
  group_by(ProteinID, peak, midpoint, experiment_name) %>%
    summarize(n = n())

all_combinations_test %>% filter(grepl("SHOT", ProteinID)) %>% select(peak) %>% unique
# Max size of contingency_full is num_midpoints * numpeaks * numprots * 4
```



```{r}
all_combinations_test <- with(contingency_full_step1_test, 
                         expand.grid(ProteinID = unique(ProteinID),
                                     experiment_name = unique(experiment_name),
                                     peak = unique(peak),
                                     midpoint = unique(midpoint),
                                     pep_half = c("firsthalf", "secondhalf"),
                                     presence = c("missing", "observed")))

all_combinations_test 

 completed <- contingency_full_step1 %>%
  group_by(ProteinID, peak, midpoint, experiment_name) %>%
   summarize(num = n())
completed %>% filter(num == 3)

```



NOT USING THIS, using script
```{r}
short_tidy_unique_peaks_manual <- short_tidy_unique_manual %>% 
  left_join(peaks_manual, by = c("ProteinID", "experiment_name", "FractionOrder")) %>%
  select(Peptide, experiment_name, ProteinID, Start, End, FractionOrder, pepcount, peak)



short_tidy_unique_peaks_manual 

short_tidy_unique_peaks_expanded_manual <-  short_tidy_unique_peaks_manual %>%
   mutate(pepid = paste0(Start, "-", End)) %>%
   mutate(presence = "observed") %>%
  
    group_by(ProteinID) %>%
     complete(nesting(pepid, Peptide), nesting(experiment_name, peak)) %>%
   ungroup %>%
     separate(pepid, into = c("Start", "End"), sep = "-", convert = TRUE, remove = FALSE) %>% # get start and end for filled in peptides
    mutate(presence = replace_na(presence, "missing")) # Filled in peptides get zero presence

# Get peaks that have at least 7 unique peptides
enough_cov_peaks_manual <- short_tidy_unique_peaks_manual %>%
   filter(!is.na(peak)) %>%
    group_by(ProteinID, peak, experiment_name) %>%
    mutate(n = n()) %>%
  ungroup %>%
  filter(n >= 7) %>%
  select(-n)

short_tidy_unique_peaks_expanded_fragments_manual  <- short_tidy_unique_peaks_expanded_manual %>%
  filter(!is.na(peak)) %>%
  semi_join(enough_cov_peaks_manual, by = c("ProteinID", "peak", "experiment_name"))
  
short_tidy_unique_peaks_expanded_fragments_manual  %>%
  write_csv("data/short_tidy_unique_peaks_expanded_fragments_manual.csv")


short_tidy_unique_peaks_expanded_fragments_manual <- read_csv("data/short_tidy_unique_peaks_expanded_fragments_manual.csv")

# about 1/20th the size of the full dataset

```

Calculate terminal bias for the manually annotated fragments
```{r}

# Take some time, est 
short_tidy_unique_peaks_expanded_fragments_tb_manual<- short_tidy_unique_peaks_expanded_fragments_manual %>%
  
  arrange(Start, End)%>%
  split(list(.$ProteinID, .$experiment_name, .$peak)) %>%
  future_map_dfr(~possibly_get_bps(.))

```



### Calculate terminal bias for each peak
```{r}

# Get which peptides were observed in each peak
obs_key_manual <- short_tidy_unique_peaks_expanded_fragments_manual %>%
  select(ProteinID, experiment_name, peak, Start, presence) %>% unique

# Need to get either next observed peptide past peak for C-terminal fragment 
# or previous observed pepide prior to peak for N-terminal fragment



short_tidy_unique_peaks_expanded_fragments_tb_manual_fc <- short_tidy_unique_peaks_expanded_fragments_tb_manual %>%
  group_by(ProteinID, experiment_name, peak) %>%
      filter(abs_log2fc == max(abs_log2fc))%>% data.frame() %>%

  left_join(obs_key_manual, by = c("ProteinID", "experiment_name", "peak")) %>%
     arrange(Start) %>%
  filter(presence == "observed") %>%
  mutate(diff = Start - bp_start) %>% 
  mutate(remove_flag = case_when(log2fc > 0 & diff >0  ~ TRUE, # Only look to left of N-term frag
                                       
                           log2fc < 0 & diff < 0 ~ TRUE,
                           TRUE ~ FALSE
                            )) %>% #Only look toward C-terminal for C-term frag
  filter(remove_flag != TRUE) %>%
  filter(diff !=0) %>% # Always offset by 1 
  arrange(diff ) %>%
  group_by(ProteinID, experiment_name, peak) %>%
    mutate(terminal = case_when(log2fc > 0 ~ "nterm",
                              log2fc < 0 ~ "cterm")) %>%
   mutate(actual_terminal_peptide = case_when(log2fc > 0 & diff == max(diff) ~ Start,
                                       
                                              log2fc < 0 & diff == min(diff) ~ Start)) %>%
  filter(!is.na(actual_terminal_peptide)) %>%
  
   # Fragments must contain at least 3 peptides. Missing ends must contain at least 3 peptides
   # Hasn't been calibrated except for nterm and missing c term
     mutate(toosmallfrag=  case_when(
     terminal == "nterm" & observed_nterm <= 3 ~ "toosmall", # Actually only 2 observed
     terminal == "cterm" & observed_cterm <= 4 ~ "toosmall",# Actually only 2 observed
     terminal == "nterm" & missing_cterm <= 3 ~ "toosmall", # Actually missing 2 peptides
     terminal == "cterm" & missing_nterm < 3 ~ "toosmall",
     TRUE  ~ "fine"
     
   )) %>%
  filter(toosmallfrag == "fine") %>%
  select(-toosmallfrag) %>% 
  unique %>%
  ungroup

short_tidy_unique_peaks_expanded_fragments_tb_manual_fc  %>%
  write_csv("data/short_tidy_unique_peaks_expanded_fragments_tb_manual_fc.csv")


# Only need the terminal bias score for the identified terminal 
```


# END unused part
